{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import subprocess\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from scipy.stats import hypergeom\n",
    "from IPython.core.display import HTML\n",
    "import pybedtools\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def annotatePeaks_stranded(peaks, motif, species=\"hg19\", hist=10, size=1000):\n",
    "    \n",
    "    \"\"\"\n",
    "    \n",
    "    runs homer with standard args\n",
    "    output location is saved\n",
    "    \n",
    "    foreground - str, location of fasta file for the foreground distribution\n",
    "    background - str, location of fasta file for the background distribution\n",
    "    k - different k-mers to examine\n",
    "    outloc - directory to output homer results \n",
    "\n",
    "    --make optional make work off locations and not fasta files \n",
    "    \n",
    "    \"\"\"\n",
    "    #findMotifs.pl clusters.fa fasta outloc -nofacts p 4 -rna -S 10 -len 5,6,7,8,9 -noconvert -nogo -fasta background.fa\n",
    "    #converts k to a string for use in subprocess\n",
    "    #I'm forgetting strand specific information here\n",
    "    try:\n",
    "        with open(\"tmp.txt\", 'w') as out:\n",
    "            subprocess.call([\"annotatePeaks.pl\",\n",
    "                             peaks,\n",
    "                             species,\n",
    "                             \"-m\", motif,\n",
    "                             #\"-hist\", str(hist),\n",
    "                             \"-size\", \"given\",\n",
    "                             \"-noann\"], shell=False, stdout=out)\n",
    "    except OSError:\n",
    "        print \"Homer not installed, ignoring motif generation, install homer for this to work\"  \n",
    "        raise   \n",
    "    annotation = pd.read_table(\"tmp.txt\", index_col=0)\n",
    "    annotation['stranded_peaks'] = annotation.apply(only_stranded_motifs, axis=1)\n",
    "    return annotation\n",
    "\n",
    "def only_stranded_motifs(row):\n",
    "    motifs = row[-1]\n",
    "    try:\n",
    "        true_peaks = \"),\".join([motif for motif in motifs.split(\"),\") if row.Strand in motif])\n",
    "        if len(true_peaks) == 0:\n",
    "            return np.nan\n",
    "        \n",
    "        return true_peaks\n",
    "    except AttributeError:\n",
    "        return motifs\n",
    "    \n",
    "def get_strand_of_motif(motifs):\n",
    "    try:\n",
    "        return \",\".join(list(set(motif.split(\",\")[1] for motif in motifs.split(\"),\"))))\n",
    "    except AttributeError:\n",
    "        return np.nan\n",
    "    \n",
    "def bound(row):\n",
    "    try:\n",
    "        return row.Strand in row.motif_strands\n",
    "    except TypeError:\n",
    "        return False\n",
    "    \n",
    "def get_bound_motifs(bedtool, motif_file, motif_name):\n",
    "    annoted_peaks = annotatePeaks(bedtool.fn, motif_file, \"mm9\")\n",
    "    annoted_peaks['motif_strands'] = annoted_peaks['1-{} Distance From Peak(sequence,strand,conservation)'.format(motif_name)].apply(get_strand_of_motif)\n",
    "    annoted_peaks['motif_present'] = annoted_peaks.apply(bound, axis=1)\n",
    "    return annoted_peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def findMotifsGenome(peaks, motif, species=\"hg19\"):\n",
    "    try:\n",
    "        with open(\"tmp.txt\", 'w') as out:\n",
    "            subprocess.call([\"findMotifsGenome.pl\",\n",
    "                             peaks,\n",
    "                             species,\n",
    "                             \"foo\",\n",
    "                             \"-size\", \"given \",\n",
    "                             \"-rna\",\n",
    "                             \"-find\", motif,\n",
    "                            ], shell=False, stdout=out)\n",
    "    except OSError:\n",
    "        print \"Homer not installed, ignoring motif generation, install homer for this to work\"  \n",
    "        raise   \n",
    "    return pd.read_table(\"tmp.txt\", index_col=0)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def findMotifsGenome(peaks, motif, species=\"hg19\"):\n",
    "    !findMotifsGenome.pl $peaks $species foo -size given -rna -find $motif > tmp.txt\n",
    "    return pd.read_table(\"tmp.txt\", index_col=0)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "rbfox2_motif = \"/home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#dir_name = \"/projects/ps-yeolab2/encode/analysis/encode_v9/assigned/\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "v12 = glob.glob(\"/projects/ps-yeolab2/encode/analysis/encode_v12/assigned/*\")\n",
    "v8 = glob.glob(\"/home/gpratt/projects/encode/analysis/encode_v8/assigned/*\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "v12_peak_sizes = {os.path.basename(item): pybedtools.BedTool(item) for item in v12}\n",
    "v8_peak_sizes = {os.path.basename(item): pybedtools.BedTool(item) for item in 8}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "dir_name = \"/home/gpratt/projects/encode/analysis/encode_v8/assigned\"\n",
    "peaks_to_annotate_v8 = {\"RBFOX2_293XT_1to200_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"RBFOX2_293XT_1to100_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"RBFOX2_293XT_1to50_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"RBFOX2_293XT_1to25_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"RBFOX2_293XT_1to12_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP10_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP8_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP1_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP5_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP6_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),}\n",
    "\n",
    "dir_name = \"/projects/ps-yeolab2/encode/analysis/encode_v12/assigned/\"\n",
    "peaks_to_annotate_v12 = {\"RBFOX2_293XT_1to200_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"RBFOX2_293XT_1to100_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"RBFOX2_293XT_1to50_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"RBFOX2_293XT_1to25_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"RBFOX2_293XT_1to12_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP10_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP8_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP1_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP5_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP6_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),}\n",
    "\n",
    "\n",
    "peaks_to_rand_1_annotate = {\"RBFOX2_293XT_1to200_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),\n",
    "                     \"RBFOX2_293XT_1to100_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),\n",
    "                     \"RBFOX2_293XT_1to50_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),\n",
    "                     \"RBFOX2_293XT_1to25_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),\n",
    "                     \"RBFOX2_293XT_1to12_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),\n",
    "                     \"EVN_041315_CLIP10_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),\n",
    "                     \"EVN_041315_CLIP8_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),\n",
    "                     \"EVN_041315_CLIP1_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),\n",
    "                     \"EVN_041315_CLIP5_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),\n",
    "                     \"EVN_041315_CLIP6_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),}\n",
    "\n",
    "peaks_to_rand_2_annotate = {\"RBFOX2_293XT_1to200_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),\n",
    "                     \"RBFOX2_293XT_1to100_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),\n",
    "                     \"RBFOX2_293XT_1to50_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),\n",
    "                     \"RBFOX2_293XT_1to25_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),\n",
    "                     \"RBFOX2_293XT_1to12_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),\n",
    "                     \"EVN_041315_CLIP10_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),\n",
    "                     \"EVN_041315_CLIP8_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),\n",
    "                     \"EVN_041315_CLIP1_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),\n",
    "                     \"EVN_041315_CLIP5_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),\n",
    "                     \"EVN_041315_CLIP6_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),}\n",
    "\n",
    "peaks_to_rand_3_annotate = {\"RBFOX2_293XT_1to200_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),\n",
    "                     \"RBFOX2_293XT_1to100_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),\n",
    "                     \"RBFOX2_293XT_1to50_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),\n",
    "                     \"RBFOX2_293XT_1to25_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),\n",
    "                     \"RBFOX2_293XT_1to12_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),\n",
    "                     \"EVN_041315_CLIP10_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),\n",
    "                     \"EVN_041315_CLIP8_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),\n",
    "                     \"EVN_041315_CLIP1_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),\n",
    "                     \"EVN_041315_CLIP5_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),\n",
    "                     \"EVN_041315_CLIP6_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "dir_name = \"/home/gpratt/projects/encode/analysis/encode_v8/\"\n",
    "peaks_to_annotate_v8 = {\"RBFOX2_293XT_1to200_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed\"),\n",
    "                     \"RBFOX2_293XT_1to100_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed\"),\n",
    "                     \"RBFOX2_293XT_1to50_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed\"),\n",
    "                     \"RBFOX2_293XT_1to25_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed\"),\n",
    "                     \"RBFOX2_293XT_1to12_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed\"),\n",
    "                     \"EVN_041315_CLIP10_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed\"),\n",
    "                     \"EVN_041315_CLIP8_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed\"),\n",
    "                     \"EVN_041315_CLIP1_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed\"),\n",
    "                     \"EVN_041315_CLIP5_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed\"),\n",
    "                     \"EVN_041315_CLIP6_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed\"),}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "dir_name = \"/projects/ps-yeolab2/encode/analysis/encode_v12/\"\n",
    "peaks_to_annotate_v12 = {\"RBFOX2_293XT_1to200_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.bam\"),\n",
    "                     \"RBFOX2_293XT_1to100_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed\"),\n",
    "                     \"RBFOX2_293XT_1to50_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed\"),\n",
    "                     \"RBFOX2_293XT_1to25_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed\"),\n",
    "                     \"RBFOX2_293XT_1to12_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed\"),\n",
    "                     \"EVN_041315_CLIP10_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed\"),\n",
    "                     \"EVN_041315_CLIP8_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed\"),\n",
    "                     \"EVN_041315_CLIP1_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed\"),\n",
    "                     \"EVN_041315_CLIP5_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed\"),\n",
    "                     \"EVN_041315_CLIP6_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed\"),}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/projects/ps-yeolab2/encode/analysis/encode_v12/EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed EVN_041315_CLIP10_RBFOX2 36715\n",
      "/projects/ps-yeolab2/encode/analysis/encode_v12/EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed EVN_041315_CLIP1_RBFOX2 29073\n",
      "/projects/ps-yeolab2/encode/analysis/encode_v12/EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed EVN_041315_CLIP5_RBFOX2 22687\n",
      "/projects/ps-yeolab2/encode/analysis/encode_v12/EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed EVN_041315_CLIP6_RBFOX2 1137\n",
      "/projects/ps-yeolab2/encode/analysis/encode_v12/EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed EVN_041315_CLIP8_RBFOX2 22113\n",
      "/projects/ps-yeolab2/encode/analysis/encode_v12/RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed RBFOX2_293XT_1to100_gel_RBFOX2 37600\n",
      "/projects/ps-yeolab2/encode/analysis/encode_v12/RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed RBFOX2_293XT_1to12_gel_RBFOX2 36222\n",
      "/projects/ps-yeolab2/encode/analysis/encode_v12/RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed RBFOX2_293XT_1to200_gel_RBFOX2 23991\n",
      "/projects/ps-yeolab2/encode/analysis/encode_v12/RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed RBFOX2_293XT_1to25_gel_RBFOX2 29041\n",
      "/projects/ps-yeolab2/encode/analysis/encode_v12/RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed RBFOX2_293XT_1to50_gel_RBFOX2 30585\n"
     ]
    }
   ],
   "source": [
    "for name, fn in sorted(peaks_to_annotate_v12.items(), key=lambda x: x[0]):\n",
    "    print fn, name, len(pybedtools.BedTool(fn))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/gpratt/projects/encode/analysis/encode_v8/EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed EVN_041315_CLIP10_RBFOX2 36715\n",
      "/home/gpratt/projects/encode/analysis/encode_v8/EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed EVN_041315_CLIP1_RBFOX2 29073\n",
      "/home/gpratt/projects/encode/analysis/encode_v8/EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed EVN_041315_CLIP5_RBFOX2 22687\n",
      "/home/gpratt/projects/encode/analysis/encode_v8/EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed EVN_041315_CLIP6_RBFOX2 1137\n",
      "/home/gpratt/projects/encode/analysis/encode_v8/EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed EVN_041315_CLIP8_RBFOX2 22113\n",
      "/home/gpratt/projects/encode/analysis/encode_v8/RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed RBFOX2_293XT_1to100_gel_RBFOX2 37600\n",
      "/home/gpratt/projects/encode/analysis/encode_v8/RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed RBFOX2_293XT_1to12_gel_RBFOX2 36222\n",
      "/home/gpratt/projects/encode/analysis/encode_v8/RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed RBFOX2_293XT_1to200_gel_RBFOX2 23991\n",
      "/home/gpratt/projects/encode/analysis/encode_v8/RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed RBFOX2_293XT_1to25_gel_RBFOX2 29041\n",
      "/home/gpratt/projects/encode/analysis/encode_v8/RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed RBFOX2_293XT_1to50_gel_RBFOX2 30585\n"
     ]
    }
   ],
   "source": [
    "for name, fn in sorted(peaks_to_annotate_v8.items(), key=lambda x: x[0]):\n",
    "    try:\n",
    "        \n",
    "        print fn, name, len(pybedtools.BedTool(fn))\n",
    "    except:\n",
    "        pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def bound_in_same_strand(row):\n",
    "    try:\n",
    "        motifs = row['1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_0.6(0.999) Distance From Peak(sequence,strand,conservation)'].split(\"),\")\n",
    "        return any(row.Strand in item.split(\",\")[1] for item in motifs)\n",
    "    except:\n",
    "        return False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# results_annotate = {}\n",
    "# for peak in peaks_to_annotate:\n",
    "#     annotated_peak = annotatePeaks_stranded(peaks_to_annotate[peak], rbfox2_motif)\n",
    "#     results_annotate[peak] = {\"peaks_with_motif\": len(annotated_peak.stranded_peaks.dropna()), \"num_peaks\": len(pybedtools.BedTool(peaks_to_annotate[peak]))}\n",
    "    \n",
    "# results_rand_1_annotate = {}\n",
    "# for peak in peaks_to_rand_1_annotate:\n",
    "#     annotated_peak = annotatePeaks_stranded(peaks_to_rand_1_annotate[peak], rbfox2_motif)\n",
    "#     #annotated_peak['bound_in_same_strand'] = annotated_peak.apply(bound_in_same_strand, axis=1)\n",
    "#     results_rand_1_annotate[peak] = {\"peaks_with_motif\": len(annotated_peak.stranded_peaks.dropna()), \"num_peaks\": len(pybedtools.BedTool(peaks_to_annotate[peak]))}\n",
    "    \n",
    "# results_rand_2_annotate = {}\n",
    "# for peak in peaks_to_rand_2_annotate:\n",
    "#     annotated_peak = annotatePeaks_stranded(peaks_to_rand_2_annotate[peak], rbfox2_motif)\n",
    "#     annotated_peak['bound_in_same_strand'] = annotated_peak.apply(bound_in_same_strand, axis=1)\n",
    "#     results_rand_2_annotate[peak] = {\"peaks_with_motif\": len(annotated_peak.stranded_peaks.dropna()), \"num_peaks\": len(pybedtools.BedTool(peaks_to_annotate[peak]))}\n",
    "#     #\n",
    "# results_rand_3_annotate = {}\n",
    "# for peak in peaks_to_rand_3_annotate:\n",
    "#     annotated_peak = annotatePeaks_stranded(peaks_to_rand_3_annotate[peak], rbfox2_motif)\n",
    "#     #annotated_peak['bound_in_same_strand'] = annotated_peak.apply(bound_in_same_strand, axis=1)\n",
    "#     results_rand_3_annotate[peak] = {\"peaks_with_motif\": len(annotated_peak.stranded_peaks.dropna()), \"num_peaks\": len(pybedtools.BedTool(peaks_to_annotate[peak]))}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 33194\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 33194\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 38 (avg size of targets)\n",
      "\tBackground files for 38 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 3573 sequences from chr1\n",
      "\tExtracting 2082 sequences from chr2\n",
      "\tExtracting 1760 sequences from chr3\n",
      "\tExtracting 892 sequences from chr4\n",
      "\tExtracting 1434 sequences from chr5\n",
      "\tExtracting 1597 sequences from chr6\n",
      "\tExtracting 1560 sequences from chr7\n",
      "\tExtracting 1000 sequences from chr8\n",
      "\tExtracting 1209 sequences from chr9\n",
      "\tExtracting 1113 sequences from chr10\n",
      "\tExtracting 1863 sequences from chr11\n",
      "\tExtracting 1975 sequences from chr12\n",
      "\tExtracting 427 sequences from chr13\n",
      "\tExtracting 1173 sequences from chr14\n",
      "\tExtracting 1041 sequences from chr15\n",
      "\tExtracting 1874 sequences from chr16\n",
      "\tExtracting 2592 sequences from chr17\n",
      "\tExtracting 414 sequences from chr18\n",
      "\tExtracting 2207 sequences from chr19\n",
      "\tExtracting 786 sequences from chr20\n",
      "\tExtracting 412 sequences from chr21\n",
      "\tExtracting 1038 sequences from chr22\n",
      "\tExtracting 1104 sequences from chrX\n",
      "\tExtracting 68 sequences from chrM\n",
      "\t511 of 33194 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t32683 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 914\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 914\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 35 (avg size of targets)\n",
      "\tBackground files for 35 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 77 sequences from chr1\n",
      "\tExtracting 47 sequences from chr2\n",
      "\tExtracting 35 sequences from chr3\n",
      "\tExtracting 33 sequences from chr4\n",
      "\tExtracting 28 sequences from chr5\n",
      "\tExtracting 30 sequences from chr6\n",
      "\tExtracting 57 sequences from chr7\n",
      "\tExtracting 35 sequences from chr8\n",
      "\tExtracting 42 sequences from chr9\n",
      "\tExtracting 53 sequences from chr10\n",
      "\tExtracting 48 sequences from chr11\n",
      "\tExtracting 33 sequences from chr12\n",
      "\tExtracting 12 sequences from chr13\n",
      "\tExtracting 27 sequences from chr14\n",
      "\tExtracting 13 sequences from chr15\n",
      "\tExtracting 59 sequences from chr16\n",
      "\tExtracting 53 sequences from chr17\n",
      "\tExtracting 11 sequences from chr18\n",
      "\tExtracting 65 sequences from chr19\n",
      "\tExtracting 40 sequences from chr20\n",
      "\tExtracting 17 sequences from chr21\n",
      "\tExtracting 40 sequences from chr22\n",
      "\tExtracting 20 sequences from chrX\n",
      "\tExtracting 39 sequences from chrM\n",
      "\t1 of 914 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t913 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 20495\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 20495\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 33 (avg size of targets)\n",
      "\tBackground files for 33 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2191 sequences from chr1\n",
      "\tExtracting 1245 sequences from chr2\n",
      "\tExtracting 1150 sequences from chr3\n",
      "\tExtracting 560 sequences from chr4\n",
      "\tExtracting 800 sequences from chr5\n",
      "\tExtracting 945 sequences from chr6\n",
      "\tExtracting 885 sequences from chr7\n",
      "\tExtracting 564 sequences from chr8\n",
      "\tExtracting 761 sequences from chr9\n",
      "\tExtracting 685 sequences from chr10\n",
      "\tExtracting 1153 sequences from chr11\n",
      "\tExtracting 1170 sequences from chr12\n",
      "\tExtracting 249 sequences from chr13\n",
      "\tExtracting 597 sequences from chr14\n",
      "\tExtracting 581 sequences from chr15\n",
      "\tExtracting 1208 sequences from chr16\n",
      "\tExtracting 1652 sequences from chr17\n",
      "\tExtracting 211 sequences from chr18\n",
      "\tExtracting 1489 sequences from chr19\n",
      "\tExtracting 541 sequences from chr20\n",
      "\tExtracting 279 sequences from chr21\n",
      "\tExtracting 757 sequences from chr22\n",
      "\tExtracting 759 sequences from chrX\n",
      "\tExtracting 63 sequences from chrM\n",
      "\t277 of 20495 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t20218 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 22370\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 22370\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 36 (avg size of targets)\n",
      "\tBackground files for 36 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2386 sequences from chr1\n",
      "\tExtracting 1238 sequences from chr2\n",
      "\tExtracting 1147 sequences from chr3\n",
      "\tExtracting 544 sequences from chr4\n",
      "\tExtracting 807 sequences from chr5\n",
      "\tExtracting 958 sequences from chr6\n",
      "\tExtracting 1023 sequences from chr7\n",
      "\tExtracting 651 sequences from chr8\n",
      "\tExtracting 860 sequences from chr9\n",
      "\tExtracting 667 sequences from chr10\n",
      "\tExtracting 1356 sequences from chr11\n",
      "\tExtracting 1188 sequences from chr12\n",
      "\tExtracting 253 sequences from chr13\n",
      "\tExtracting 697 sequences from chr14\n",
      "\tExtracting 555 sequences from chr15\n",
      "\tExtracting 1456 sequences from chr16\n",
      "\tExtracting 1970 sequences from chr17\n",
      "\tExtracting 218 sequences from chr18\n",
      "\tExtracting 1831 sequences from chr19\n",
      "\tExtracting 636 sequences from chr20\n",
      "\tExtracting 308 sequences from chr21\n",
      "\tExtracting 879 sequences from chr22\n",
      "\tExtracting 701 sequences from chrX\n",
      "\tExtracting 41 sequences from chrM\n",
      "\t265 of 22370 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t22105 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 34934\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 34934\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 38 (avg size of targets)\n",
      "\tBackground files for 38 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 3680 sequences from chr1\n",
      "\tExtracting 2093 sequences from chr2\n",
      "\tExtracting 1870 sequences from chr3\n",
      "\tExtracting 886 sequences from chr4\n",
      "\tExtracting 1304 sequences from chr5\n",
      "\tExtracting 1539 sequences from chr6\n",
      "\tExtracting 1591 sequences from chr7\n",
      "\tExtracting 1063 sequences from chr8\n",
      "\tExtracting 1301 sequences from chr9\n",
      "\tExtracting 1063 sequences from chr10\n",
      "\tExtracting 2045 sequences from chr11\n",
      "\tExtracting 1990 sequences from chr12\n",
      "\tExtracting 460 sequences from chr13\n",
      "\tExtracting 1147 sequences from chr14\n",
      "\tExtracting 1047 sequences from chr15\n",
      "\tExtracting 2210 sequences from chr16\n",
      "\tExtracting 2958 sequences from chr17\n",
      "\tExtracting 419 sequences from chr18\n",
      "\tExtracting 2522 sequences from chr19\n",
      "\tExtracting 875 sequences from chr20\n",
      "\tExtracting 427 sequences from chr21\n",
      "\tExtracting 1252 sequences from chr22\n",
      "\tExtracting 1145 sequences from chrX\n",
      "\tExtracting 47 sequences from chrM\n",
      "\t419 of 34934 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t34515 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 27038\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 27038\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 31 (avg size of targets)\n",
      "\tBackground files for 31 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2806 sequences from chr1\n",
      "\tExtracting 1609 sequences from chr2\n",
      "\tExtracting 1452 sequences from chr3\n",
      "\tExtracting 751 sequences from chr4\n",
      "\tExtracting 978 sequences from chr5\n",
      "\tExtracting 1211 sequences from chr6\n",
      "\tExtracting 1291 sequences from chr7\n",
      "\tExtracting 760 sequences from chr8\n",
      "\tExtracting 1130 sequences from chr9\n",
      "\tExtracting 925 sequences from chr10\n",
      "\tExtracting 1442 sequences from chr11\n",
      "\tExtracting 1359 sequences from chr12\n",
      "\tExtracting 424 sequences from chr13\n",
      "\tExtracting 760 sequences from chr14\n",
      "\tExtracting 786 sequences from chr15\n",
      "\tExtracting 1571 sequences from chr16\n",
      "\tExtracting 2054 sequences from chr17\n",
      "\tExtracting 347 sequences from chr18\n",
      "\tExtracting 2101 sequences from chr19\n",
      "\tExtracting 816 sequences from chr20\n",
      "\tExtracting 423 sequences from chr21\n",
      "\tExtracting 1114 sequences from chr22\n",
      "\tExtracting 877 sequences from chrX\n",
      "\tExtracting 51 sequences from chrM\n",
      "\t255 of 27038 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t26783 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 33665\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 33665\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 41 (avg size of targets)\n",
      "\tBackground files for 41 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 3475 sequences from chr1\n",
      "\tExtracting 1969 sequences from chr2\n",
      "\tExtracting 1793 sequences from chr3\n",
      "\tExtracting 899 sequences from chr4\n",
      "\tExtracting 1280 sequences from chr5\n",
      "\tExtracting 1362 sequences from chr6\n",
      "\tExtracting 1600 sequences from chr7\n",
      "\tExtracting 1056 sequences from chr8\n",
      "\tExtracting 1349 sequences from chr9\n",
      "\tExtracting 1069 sequences from chr10\n",
      "\tExtracting 1956 sequences from chr11\n",
      "\tExtracting 1706 sequences from chr12\n",
      "\tExtracting 437 sequences from chr13\n",
      "\tExtracting 1020 sequences from chr14\n",
      "\tExtracting 966 sequences from chr15\n",
      "\tExtracting 2149 sequences from chr16\n",
      "\tExtracting 2802 sequences from chr17\n",
      "\tExtracting 415 sequences from chr18\n",
      "\tExtracting 2622 sequences from chr19\n",
      "\tExtracting 922 sequences from chr20\n",
      "\tExtracting 466 sequences from chr21\n",
      "\tExtracting 1309 sequences from chr22\n",
      "\tExtracting 1003 sequences from chrX\n",
      "\tExtracting 40 sequences from chrM\n",
      "\t277 of 33665 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t33388 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 28458\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 28458\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 41 (avg size of targets)\n",
      "\tBackground files for 41 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2986 sequences from chr1\n",
      "\tExtracting 1655 sequences from chr2\n",
      "\tExtracting 1472 sequences from chr3\n",
      "\tExtracting 740 sequences from chr4\n",
      "\tExtracting 1065 sequences from chr5\n",
      "\tExtracting 1208 sequences from chr6\n",
      "\tExtracting 1317 sequences from chr7\n",
      "\tExtracting 848 sequences from chr8\n",
      "\tExtracting 1088 sequences from chr9\n",
      "\tExtracting 863 sequences from chr10\n",
      "\tExtracting 1658 sequences from chr11\n",
      "\tExtracting 1496 sequences from chr12\n",
      "\tExtracting 348 sequences from chr13\n",
      "\tExtracting 867 sequences from chr14\n",
      "\tExtracting 750 sequences from chr15\n",
      "\tExtracting 1906 sequences from chr16\n",
      "\tExtracting 2406 sequences from chr17\n",
      "\tExtracting 341 sequences from chr18\n",
      "\tExtracting 2245 sequences from chr19\n",
      "\tExtracting 789 sequences from chr20\n",
      "\tExtracting 374 sequences from chr21\n",
      "\tExtracting 1103 sequences from chr22\n",
      "\tExtracting 887 sequences from chrX\n",
      "\tExtracting 46 sequences from chrM\n",
      "\t270 of 28458 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t28188 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 20718\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 20718\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 33 (avg size of targets)\n",
      "\tBackground files for 33 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2019 sequences from chr1\n",
      "\tExtracting 1258 sequences from chr2\n",
      "\tExtracting 1165 sequences from chr3\n",
      "\tExtracting 621 sequences from chr4\n",
      "\tExtracting 721 sequences from chr5\n",
      "\tExtracting 958 sequences from chr6\n",
      "\tExtracting 988 sequences from chr7\n",
      "\tExtracting 604 sequences from chr8\n",
      "\tExtracting 882 sequences from chr9\n",
      "\tExtracting 760 sequences from chr10\n",
      "\tExtracting 1026 sequences from chr11\n",
      "\tExtracting 876 sequences from chr12\n",
      "\tExtracting 400 sequences from chr13\n",
      "\tExtracting 575 sequences from chr14\n",
      "\tExtracting 556 sequences from chr15\n",
      "\tExtracting 1041 sequences from chr16\n",
      "\tExtracting 1568 sequences from chr17\n",
      "\tExtracting 274 sequences from chr18\n",
      "\tExtracting 1710 sequences from chr19\n",
      "\tExtracting 654 sequences from chr20\n",
      "\tExtracting 341 sequences from chr21\n",
      "\tExtracting 946 sequences from chr22\n",
      "\tExtracting 733 sequences from chrX\n",
      "\tExtracting 42 sequences from chrM\n",
      "\t118 of 20718 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t20600 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 27088\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 27088\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 38 (avg size of targets)\n",
      "\tBackground files for 38 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2873 sequences from chr1\n",
      "\tExtracting 1574 sequences from chr2\n",
      "\tExtracting 1370 sequences from chr3\n",
      "\tExtracting 708 sequences from chr4\n",
      "\tExtracting 1001 sequences from chr5\n",
      "\tExtracting 1125 sequences from chr6\n",
      "\tExtracting 1287 sequences from chr7\n",
      "\tExtracting 783 sequences from chr8\n",
      "\tExtracting 1094 sequences from chr9\n",
      "\tExtracting 849 sequences from chr10\n",
      "\tExtracting 1541 sequences from chr11\n",
      "\tExtracting 1420 sequences from chr12\n",
      "\tExtracting 326 sequences from chr13\n",
      "\tExtracting 826 sequences from chr14\n",
      "\tExtracting 713 sequences from chr15\n",
      "\tExtracting 1772 sequences from chr16\n",
      "\tExtracting 2280 sequences from chr17\n",
      "\tExtracting 271 sequences from chr18\n",
      "\tExtracting 2177 sequences from chr19\n",
      "\tExtracting 780 sequences from chr20\n",
      "\tExtracting 381 sequences from chr21\n",
      "\tExtracting 1081 sequences from chr22\n",
      "\tExtracting 819 sequences from chrX\n",
      "\tExtracting 37 sequences from chrM\n",
      "\t239 of 27088 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t26849 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "results_v8 = {}\n",
    "for peak in peaks_to_annotate_v8:\n",
    "    annotated_peak = findMotifsGenome(peaks_to_annotate[peak], rbfox2_motif)\n",
    "    results[peak] = {\"peaks_with_motif\": len(set(annotated_peak.index)), \"num_peaks\": len(pybedtools.BedTool(peaks_to_annotate[peak]))}\n",
    " \n",
    "results_v12 = {}\n",
    "for peak in peaks_to_annotate_v12:\n",
    "    annotated_peak = findMotifsGenome(peaks_to_annotate[peak], rbfox2_motif)\n",
    "    results[peak] = {\"peaks_with_motif\": len(set(annotated_peak.index)), \"num_peaks\": len(pybedtools.BedTool(peaks_to_annotate[peak]))}\n",
    "    \n",
    "    \n",
    "# results_rand_1 = {}\n",
    "# for peak in peaks_to_rand_1_annotate:\n",
    "#     annotated_peak = findMotifsGenome(peaks_to_rand_1_annotate[peak], rbfox2_motif)\n",
    "#     annotated_peak['bound_in_same_strand'] = annotated_peak.apply(bound_in_same_strand, axis=1)\n",
    "#     results_rand_1[peak] = {\"peaks_with_motif\": len(set(annotated_peak.index)), \"num_peaks\": len(pybedtools.BedTool(peaks_to_annotate[peak]))}\n",
    "    \n",
    "# results_rand_2 = {}\n",
    "# for peak in peaks_to_rand_2_annotate:\n",
    "#     annotated_peak = findMotifsGenome(peaks_to_rand_2_annotate[peak], rbfox2_motif)\n",
    "#     annotated_peak['bound_in_same_strand'] = annotated_peak.apply(bound_in_same_strand, axis=1)\n",
    "#     results_rand_2[peak] = {\"peaks_with_motif\": len(set(annotated_peak.index)), \"num_peaks\": len(pybedtools.BedTool(peaks_to_annotate[peak]))}\n",
    "    \n",
    "# results_rand_3 = {}\n",
    "# for peak in peaks_to_rand_3_annotate:\n",
    "#     annotated_peak = findMotifsGenome(peaks_to_rand_3_annotate[peak], rbfox2_motif)\n",
    "#     annotated_peak['bound_in_same_strand'] = annotated_peak.apply(bound_in_same_strand, axis=1)\n",
    "#     results_rand_3[peak] = {\"peaks_with_motif\": len(set(annotated_peak.index)), \"num_peaks\": len(pybedtools.BedTool(peaks_to_annotate[peak]))}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "real = pd.DataFrame(results).T\n",
    "\n",
    "random = pd.concat([pd.DataFrame(results_rand_1).T, \n",
    "           pd.DataFrame(results_rand_2).T, \n",
    "           pd.DataFrame(results_rand_3).T,])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "real.to_csv(\"/home/gpratt/Dropbox/encode_integration/for_eric/real_counts_v12.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>num_peaks</th>\n",
       "      <th>peaks_with_motif</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>EVN_041315_CLIP10_RBFOX2</th>\n",
       "      <td>33194</td>\n",
       "      <td>2178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EVN_041315_CLIP1_RBFOX2</th>\n",
       "      <td>27038</td>\n",
       "      <td>2420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EVN_041315_CLIP5_RBFOX2</th>\n",
       "      <td>20718</td>\n",
       "      <td>4011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EVN_041315_CLIP6_RBFOX2</th>\n",
       "      <td>914</td>\n",
       "      <td>260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EVN_041315_CLIP8_RBFOX2</th>\n",
       "      <td>20495</td>\n",
       "      <td>1165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBFOX2_293XT_1to100_gel_RBFOX2</th>\n",
       "      <td>34934</td>\n",
       "      <td>1945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBFOX2_293XT_1to12_gel_RBFOX2</th>\n",
       "      <td>33665</td>\n",
       "      <td>3056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBFOX2_293XT_1to200_gel_RBFOX2</th>\n",
       "      <td>22370</td>\n",
       "      <td>1023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBFOX2_293XT_1to25_gel_RBFOX2</th>\n",
       "      <td>27088</td>\n",
       "      <td>1821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBFOX2_293XT_1to50_gel_RBFOX2</th>\n",
       "      <td>28458</td>\n",
       "      <td>1898</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                num_peaks  peaks_with_motif\n",
       "EVN_041315_CLIP10_RBFOX2            33194              2178\n",
       "EVN_041315_CLIP1_RBFOX2             27038              2420\n",
       "EVN_041315_CLIP5_RBFOX2             20718              4011\n",
       "EVN_041315_CLIP6_RBFOX2               914               260\n",
       "EVN_041315_CLIP8_RBFOX2             20495              1165\n",
       "RBFOX2_293XT_1to100_gel_RBFOX2      34934              1945\n",
       "RBFOX2_293XT_1to12_gel_RBFOX2       33665              3056\n",
       "RBFOX2_293XT_1to200_gel_RBFOX2      22370              1023\n",
       "RBFOX2_293XT_1to25_gel_RBFOX2       27088              1821\n",
       "RBFOX2_293XT_1to50_gel_RBFOX2       28458              1898"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#v12\n",
    "real"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>num_peaks</th>\n",
       "      <th>peaks_with_motif</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>EVN_041315_CLIP10_RBFOX2</th>\n",
       "      <td>33194</td>\n",
       "      <td>2178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EVN_041315_CLIP1_RBFOX2</th>\n",
       "      <td>27038</td>\n",
       "      <td>2420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EVN_041315_CLIP5_RBFOX2</th>\n",
       "      <td>20718</td>\n",
       "      <td>4011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EVN_041315_CLIP6_RBFOX2</th>\n",
       "      <td>914</td>\n",
       "      <td>260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EVN_041315_CLIP8_RBFOX2</th>\n",
       "      <td>20495</td>\n",
       "      <td>1165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBFOX2_293XT_1to100_gel_RBFOX2</th>\n",
       "      <td>34934</td>\n",
       "      <td>1945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBFOX2_293XT_1to12_gel_RBFOX2</th>\n",
       "      <td>33665</td>\n",
       "      <td>3056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBFOX2_293XT_1to200_gel_RBFOX2</th>\n",
       "      <td>22370</td>\n",
       "      <td>1023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBFOX2_293XT_1to25_gel_RBFOX2</th>\n",
       "      <td>27088</td>\n",
       "      <td>1821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBFOX2_293XT_1to50_gel_RBFOX2</th>\n",
       "      <td>28458</td>\n",
       "      <td>1898</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                num_peaks  peaks_with_motif\n",
       "EVN_041315_CLIP10_RBFOX2            33194              2178\n",
       "EVN_041315_CLIP1_RBFOX2             27038              2420\n",
       "EVN_041315_CLIP5_RBFOX2             20718              4011\n",
       "EVN_041315_CLIP6_RBFOX2               914               260\n",
       "EVN_041315_CLIP8_RBFOX2             20495              1165\n",
       "RBFOX2_293XT_1to100_gel_RBFOX2      34934              1945\n",
       "RBFOX2_293XT_1to12_gel_RBFOX2       33665              3056\n",
       "RBFOX2_293XT_1to200_gel_RBFOX2      22370              1023\n",
       "RBFOX2_293XT_1to25_gel_RBFOX2       27088              1821\n",
       "RBFOX2_293XT_1to50_gel_RBFOX2       28458              1898"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#v8\n",
    "real"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "random.groupby(level=0).sum().to_csv(\"/home/gpratt/Dropbox/encode_integration/for_eric/random_counts_v12.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>num_peaks</th>\n",
       "      <th>peaks_with_motif</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>EVN_041315_CLIP10_RBFOX2</th>\n",
       "      <td>99582</td>\n",
       "      <td>1217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EVN_041315_CLIP1_RBFOX2</th>\n",
       "      <td>81114</td>\n",
       "      <td>782</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EVN_041315_CLIP5_RBFOX2</th>\n",
       "      <td>62154</td>\n",
       "      <td>649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EVN_041315_CLIP6_RBFOX2</th>\n",
       "      <td>2742</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EVN_041315_CLIP8_RBFOX2</th>\n",
       "      <td>61485</td>\n",
       "      <td>583</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBFOX2_293XT_1to100_gel_RBFOX2</th>\n",
       "      <td>104802</td>\n",
       "      <td>1214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBFOX2_293XT_1to12_gel_RBFOX2</th>\n",
       "      <td>100995</td>\n",
       "      <td>1267</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBFOX2_293XT_1to200_gel_RBFOX2</th>\n",
       "      <td>67110</td>\n",
       "      <td>730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBFOX2_293XT_1to25_gel_RBFOX2</th>\n",
       "      <td>81264</td>\n",
       "      <td>955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBFOX2_293XT_1to50_gel_RBFOX2</th>\n",
       "      <td>85374</td>\n",
       "      <td>1099</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                num_peaks  peaks_with_motif\n",
       "EVN_041315_CLIP10_RBFOX2            99582              1217\n",
       "EVN_041315_CLIP1_RBFOX2             81114               782\n",
       "EVN_041315_CLIP5_RBFOX2             62154               649\n",
       "EVN_041315_CLIP6_RBFOX2              2742                35\n",
       "EVN_041315_CLIP8_RBFOX2             61485               583\n",
       "RBFOX2_293XT_1to100_gel_RBFOX2     104802              1214\n",
       "RBFOX2_293XT_1to12_gel_RBFOX2      100995              1267\n",
       "RBFOX2_293XT_1to200_gel_RBFOX2      67110               730\n",
       "RBFOX2_293XT_1to25_gel_RBFOX2       81264               955\n",
       "RBFOX2_293XT_1to50_gel_RBFOX2       85374              1099"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "random.groupby(level=0).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
