{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import urllib\n",
    "import pandas as pd\n",
    "from gscripts.general import region_helpers\n",
    "\n",
    "\n",
    "import gffutils\n",
    "\n",
    "from clipper.src import get_genomic_regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "\"\"\"\n",
    "\n",
    "class to get genomic features from gffutils _db\n",
    "\n",
    "\"\"\"\n",
    "\n",
    "import copy\n",
    "from collections import defaultdict\n",
    "import os\n",
    "\n",
    "\n",
    "import gffutils\n",
    "import pybedtools\n",
    "from clipper import data_dir\n",
    "\n",
    "\n",
    "class GenomicFeatures():\n",
    "    \"\"\"\n",
    "\n",
    "    class to get genomic features from gffutils _db\n",
    "    \n",
    "    \"\"\"\n",
    "    def __init__(self, species, db=None, regions_dir=None, gencode=False, expressed_transcripts=None):\n",
    "        \"\"\"\n",
    "        \n",
    "        creates genomic features function, chooses \n",
    "        how to direct creation of features based on _species\n",
    "        \n",
    "        regions_dir : str location to create region\n",
    "        species: str species (hg19, mm9, ce10\n",
    "        db: gffutils FeatureDb object\n",
    "        \n",
    "        \"\"\"\n",
    "\n",
    "        if regions_dir == None:\n",
    "            regions_dir = os.path.join(data_dir(), \"regions\")\n",
    "        self._regions_dir = regions_dir\n",
    "        self._db = db\n",
    "        self._species = species\n",
    "\n",
    "        #I'm going to be lazy and leave this here, its needed to make a new genomic features for human genomes\n",
    "        #engineering so it doesn't take too much time on load will be slightly annoying so just uncomment when you need it\n",
    "#         result = defaultdict(lambda: defaultdict(lambda: defaultdict(list)))\n",
    "#         for feature in db.all_features():\n",
    "#             gene_ids = feature.attributes['gene_id']\n",
    "#             transcript_ids = feature.attributes['transcript_id']\n",
    "#             feature_type = feature.featuretype\n",
    "        \n",
    "        \n",
    "#             if feature_type == \"gene\":\n",
    "#                 if len(gene_ids) != 1:\n",
    "#                     print gene_ids[0]\n",
    "#                     break\n",
    "        \n",
    "#                 result[gene_ids[0]]['gene'] = feature\n",
    "#             else:\n",
    "#                 for gene_id in gene_ids:\n",
    "#                     for transcript_id in transcript_ids:\n",
    "#                         if transcript_id in expressed_transcripts:\n",
    "#                             result[gene_id][transcript_id][feature_type].append(feature)\n",
    "        \n",
    "#         self._feature_hash = result\n",
    "\n",
    "        if species in [\"hg19\", \"mm9\", \"hg19_v19\", \"GRCh38_v24\"] or gencode:\n",
    "            self._feature_names = {\n",
    "                             \"five_prime_utr\" : \"five_prime_utr\",\n",
    "                             \"three_prime_utr\" : \"three_prime_utr\",\n",
    "                             \"exon\" : \"exon\",\n",
    "                             \"CDS\" : \"CDS\",\n",
    "                             \"gene_id\" : \"gene_id\",\n",
    "                             \"transcript\" : \"transcript\",\n",
    "                             }\n",
    "            self._fix_chrom = self._fix_chrom_null\n",
    "            \n",
    "        elif species in [\"ce10\"]:\n",
    "            self._feature_names = {\n",
    "                             \"five_prime_utr\" : \"five_prime_UTR\",\n",
    "                             \"three_prime_utr\" : \"three_prime_UTR\",\n",
    "                             \"exon\" : \"exon\",\n",
    "                             \"CDS\" : \"CDS\",\n",
    "                             \"gene_id\" : \"ID\",\n",
    "                             \"transcript\" : \"mRNA\",\n",
    "                             }\n",
    "            self._fix_chrom = self._fix_chrom_ce10\n",
    "        if self._db is not None:\n",
    "            self.featuretypes = list(self._db.featuretypes())\n",
    "        else:\n",
    "            self.featuretypes = None\n",
    "    def _to_bed(self, interval):\n",
    "        \"\"\"\n",
    "        interval: gffutils interval\n",
    "        \n",
    "        converts gffutils to bed format, setting id to be the \n",
    "        id of the gene\n",
    "        \n",
    "        \"\"\"\n",
    "        \n",
    "        try:\n",
    "            gene_id = interval.attributes[self._feature_names['gene_id']]\n",
    "        except:\n",
    "            print type(interval), interval\n",
    "            return None\n",
    "        \n",
    "        if type(gene_id) is list:\n",
    "            gene_id = gene_id[0]\n",
    "\n",
    "        if interval.start > interval.stop:\n",
    "            print interval\n",
    "        return (interval.chrom, interval.start, \n",
    "                interval.stop, gene_id, \"0\", interval.strand)\n",
    "    \n",
    "    def _fix_chrom_ce10(self, interval):\n",
    "        \"\"\"\n",
    "        \n",
    "        Adjusts ce10 interval to be in line with ucsc browser format\n",
    "        \n",
    "        \"\"\"\n",
    "        interval.chrom = \"chr\" + interval.chrom.replace(\"tDNA\", \"\")\n",
    "        return interval \n",
    "    \n",
    "    def _fix_chrom_null(self, interval):\n",
    "        \n",
    "        \"\"\"\n",
    "        \n",
    "        null fixer for when things are in ucsc format\n",
    "        \n",
    "        \"\"\"\n",
    "        return interval\n",
    "    \n",
    "    def _get_proximal_distal_introns(self, introns, prox_size=500):\n",
    "        \"\"\"\n",
    "    \n",
    "        From a given gene returns all its proximal and distal introns, \n",
    "        proximal and distal being defined by prox_size \n",
    "        gene = iterator of exons belonging to a single gene\n",
    "        prox_size = int size of proximal introns\n",
    "    \n",
    "        returns bedtool(proximal introns), bedtool(distal introns)\n",
    "    \n",
    "        \"\"\"\n",
    "    \n",
    "        prox_introns = []\n",
    "        dist_introns = []\n",
    "\n",
    "        for intron in introns:\n",
    "            #want distal introns to have size at least one, otherwise\n",
    "            if len(intron) <= (prox_size * 2) + 3: \n",
    "                #they are proximal\n",
    "                prox_introns.append(intron)\n",
    "            else:\n",
    "                #create prox and dist intron ranges from intron \n",
    "                #(this is dangerous, but copying doesn't work\n",
    "                start_prox_intron = copy.deepcopy(intron)\n",
    "                start_prox_intron.stop = intron.start + prox_size\n",
    "                prox_introns.append(start_prox_intron)\n",
    "            \n",
    "                stop_prox_intron = copy.deepcopy(intron)\n",
    "                stop_prox_intron.start = intron.stop - prox_size\n",
    "                prox_introns.append(stop_prox_intron)\n",
    "            \n",
    "                dist_intron = copy.deepcopy(intron)\n",
    "                dist_intron.start = intron.start + prox_size + 1\n",
    "                dist_intron.stop = intron.stop - prox_size - 1\n",
    "                dist_introns.append(dist_intron)\n",
    "        return prox_introns, dist_introns\n",
    "\n",
    "    def _rename_regions(self, intervals, gene_id):\n",
    "        \"\"\"\n",
    "        \n",
    "        renames list of intervals\n",
    "        \n",
    "        intervals: gffutils interval list\n",
    "        gene_id: value to replace id with\n",
    "        \n",
    "        returns updated list of intervals\n",
    "        \"\"\"\n",
    "        updated_regions = []\n",
    "        for interval in intervals:\n",
    "            interval.attributes[self._feature_names['gene_id']] = gene_id\n",
    "            updated_regions.append(interval)\n",
    "        return updated_regions\n",
    "    \n",
    "    def _merge_and_rename_regions(self, regions, gene_id):\n",
    "        \"\"\"\n",
    "        \n",
    "        region: list gffutils features to merge \n",
    "        gene_id: str string to rename gene_id as (changes feature names to be sane)\n",
    "        \n",
    "        \"\"\"\n",
    "        \n",
    "        if len(regions) > 0:\n",
    "            regions = self._db.merge(sorted(list(regions), \n",
    "                                            key = lambda x: x.start)) \n",
    "                                           \n",
    "        \n",
    "            return self._rename_regions(regions, gene_id)\n",
    "        return []\n",
    "    \n",
    "    def _get_utrs(self, gene, mrna, cds, feature_types):\n",
    "        \"\"\"\n",
    "        \n",
    "        mrna: gffutils feature an mRNA\n",
    "        cds: list[gffutls features] coding sequences belonging to the mRNA\n",
    "        features_types: list[str] features avaiable for parsing \n",
    "        \n",
    "        returns back list[gffutils feature] for 5' and 3' UTRS\n",
    "        \"\"\"\n",
    "        mrna_five_prime_utrs = []\n",
    "        mrna_three_prime_utrs = []\n",
    "        if (self._feature_names['five_prime_utr'] in feature_types and self._feature_names['three_prime_utr'] in feature_types):\n",
    "            \n",
    "            mrna_five_prime_utrs = (list(self._db.children(mrna, featuretype=self._feature_names['five_prime_utr'])))\n",
    "            mrna_three_prime_utrs = (list(self._db.children(mrna, featuretype=self._feature_names['three_prime_utr'])))\n",
    "        else:\n",
    "            #If 5' and 3' utr annotations don't exist \n",
    "            #generate them from CDS and UTR information (handles gencode case)\n",
    "            utrs = list(self._feature_hash[gene][mrna]['UTR'])\n",
    "            if len(cds) == 0:\n",
    "                return [], []\n",
    "                \n",
    "            first_cds, last_cds = cds[0], cds[-1]\n",
    "            for utr in utrs:\n",
    "                if utr.strand == \"+\":\n",
    "                    if utr.stop < first_cds.start:\n",
    "                        utr.featuretype = self._feature_names[\"five_prime_utr\"]\n",
    "                        mrna_five_prime_utrs.append(utr)\n",
    "                    elif last_cds.stop < utr.start:\n",
    "                        utr.featuretype = self._feature_names[\"three_prime_utr\"]\n",
    "                        mrna_three_prime_utrs.append(utr)\n",
    "                    else:\n",
    "                        print \"something odd\"\n",
    "    \n",
    "                elif utr.strand == \"-\":\n",
    "                    if last_cds.stop < utr.start:\n",
    "                        utr.featuretype = self._feature_names[\"five_prime_utr\"]\n",
    "                        mrna_five_prime_utrs.append(utr)\n",
    "                    elif utr.start < first_cds.start:\n",
    "                        utr.featuretype = self._feature_names[\"three_prime_utr\"]\n",
    "                        mrna_three_prime_utrs.append(utr)\n",
    "                    else:\n",
    "                        print \"odd in the negative strand\"\n",
    "        return mrna_five_prime_utrs, mrna_three_prime_utrs\n",
    "\n",
    "    def _interval_key(self, interval):\n",
    "        return interval.start\n",
    "\n",
    "    def _gene_regions(self, gene_id, prox_size=500):\n",
    "        gene_five_prime_utrs = []\n",
    "        gene_three_prime_utrs = []\n",
    "        gene_cds = []\n",
    "        gene_exons = []\n",
    "        gene_introns = []\n",
    "        gene_dist_introns = []\n",
    "        gene_prox_introns = []\n",
    "        for mrna in self._feature_hash[gene_id].keys():\n",
    "            if mrna == \"gene\":\n",
    "                continue\n",
    "            mrna_cds = []\n",
    "            if self._feature_names['CDS'] in self._feature_hash[gene_id][mrna]:\n",
    "                mrna_cds = list(self._feature_hash[gene_id][mrna][self._feature_names['CDS']])\n",
    "            mrna_exons = []\n",
    "            if self._feature_names['exon'] in self._feature_hash[gene_id][mrna]:\n",
    "                mrna_exons = list(self._feature_hash[gene_id][mrna][self._feature_names['exon']])\n",
    "            gene_exons += mrna_exons\n",
    "            gene_cds += mrna_cds\n",
    "            mrna_five_prime_utrs, mrna_three_prime_utrs = self._get_utrs(gene_id, mrna, mrna_cds, self.featuretypes)\n",
    "            gene_five_prime_utrs += mrna_five_prime_utrs\n",
    "            gene_three_prime_utrs += mrna_three_prime_utrs\n",
    "            mrna_introns = list(self._db.interfeatures(self._db.merge(sorted(mrna_exons, key=self._interval_key))))\n",
    "            gene_introns += mrna_introns\n",
    "            mrna_prox_introns, mrna_dist_introns = self._get_proximal_distal_introns(mrna_introns, prox_size)\n",
    "            gene_dist_introns += mrna_dist_introns\n",
    "            gene_prox_introns += mrna_prox_introns\n",
    "\n",
    "        gene = self._feature_hash[gene_id]['gene']\n",
    "        #gene_id = gene.attributes[self._feature_names['gene_id']]\n",
    "\n",
    "\n",
    "\n",
    "        return self._merge_and_rename_regions(gene_cds, gene_id), \\\n",
    "               self._merge_and_rename_regions(gene_dist_introns, gene_id), \\\n",
    "               self._merge_and_rename_regions(gene_exons, gene_id),  \\\n",
    "               self._merge_and_rename_regions(gene_five_prime_utrs, gene_id), \\\n",
    "               self._merge_and_rename_regions(gene_introns, gene_id), \\\n",
    "               self._merge_and_rename_regions(gene_prox_introns, gene_id), \\\n",
    "               self._merge_and_rename_regions(gene_three_prime_utrs, gene_id)\n",
    "\n",
    "    def get_genomic_regions(self, prox_size=500, limit_genes=False, flush_cashe=False):\n",
    "        \n",
    "        \"\"\"\n",
    "        \n",
    "        returns bedtool of all non-overlapping regions in the genome, exons, cds, 3' utrs and 5' utrs\n",
    "        _species - string of the _species to analyze\n",
    "        _db - _db handle generated by gtf utils\n",
    "        \n",
    "        Potental off by one bug here, need to examine more closely\n",
    "        \"\"\"\n",
    "        region_and_species = os.path.join(self._regions_dir, self._species)\n",
    "        regions = [\"genes\", \"five_prime_utrs\", \"three_prime_utrs\", \"cds\", \n",
    "                   \"exons\", \"introns\", \"proxintron\", \"distintron\",\n",
    "                   ]\n",
    "        try:\n",
    "            if flush_cashe:\n",
    "                raise ValueError\n",
    "            results = {}\n",
    "            for region in regions:\n",
    "                if region in [\"proxintron\", \"distintron\"]:\n",
    "                    results[region] = pybedtools.BedTool(\"%s_%s%d.bed\" % (region_and_species, \n",
    "                                                                           region, prox_size))\n",
    "                else:\n",
    "                    results[region] = pybedtools.BedTool(\"%s_%s.bed\" % (region_and_species, \n",
    "                                                                         region))\n",
    "            return results\n",
    "        except ValueError as e:\n",
    "            print e\n",
    "            pass\n",
    "        \n",
    "        three_prime_utrs = []\n",
    "        five_prime_utrs = []\n",
    "        cds = []\n",
    "        exons = []\n",
    "        dist_introns = []\n",
    "        prox_introns = []\n",
    "        gene_list = []\n",
    "        introns = []\n",
    "        for i, gene in enumerate(self._feature_hash.keys()):\n",
    "            gene_list.append(self._feature_hash[gene]['gene'])\n",
    "            if i % 2000 == 0:\n",
    "                print \"processed %d genes\" % (i)\n",
    "                if i == 2000 and limit_genes:\n",
    "                    break\n",
    "\n",
    "\n",
    "            gene_cds, gene_dist_introns, gene_exons, gene_five_prime_utrs, gene_introns, gene_prox_introns, gene_three_prime_utrs = self._gene_regions(gene)\n",
    "            three_prime_utrs += gene_three_prime_utrs\n",
    "            five_prime_utrs += gene_five_prime_utrs\n",
    "            cds += gene_cds\n",
    "            exons += gene_exons\n",
    "            dist_introns += gene_dist_introns\n",
    "            prox_introns += gene_prox_introns\n",
    "            introns += gene_introns\n",
    "\n",
    "        #make exons and introns\n",
    "        results = {\"genes\": gene_list,\n",
    "                   \"five_prime_utrs\": five_prime_utrs,\n",
    "                   \"three_prime_utrs\": three_prime_utrs,\n",
    "                   \"cds\": cds,\n",
    "                   \"proxintron\": prox_introns,\n",
    "                   \"distintron\": dist_introns,\n",
    "                   \"exons\": exons,\n",
    "                   \"introns\": introns}\n",
    "        \n",
    "        for name, intervals in results.items():\n",
    "            print name\n",
    "\n",
    "            intervals = pybedtools.BedTool(map(self._to_bed, intervals)).remove_invalid().sort().each(self._fix_chrom)\n",
    "            \n",
    "            if name in [\"proxintron\", \"distintron\"]:\n",
    "                results[name] = intervals.saveas(region_and_species + \"_%s%d.bed\" % (name, \n",
    "                                                                                     prox_size))\n",
    "            else:\n",
    "                results[name] = intervals.saveas(region_and_species + \"_%s.bed\" % (name))\n",
    "\n",
    "        return results\n",
    "        \n",
    "    def get_feature_locations(self, limit_genes=False, flush_cashe=False):\n",
    "\n",
    "        \n",
    "        \"\"\"\n",
    "        \n",
    "        Gets locations of genic features, five prime sites, 3 prime sites, poly a sites stop codons start codons and tss\n",
    "        based off annotated gtf _db file\n",
    "        \n",
    "        _db - _db handle generated by gtf utils\n",
    "        \n",
    "        returns dict of bedfiles     { five_prime_ends : bedtool \n",
    "                                       three_prime_ends\n",
    "                                       poly_a_sites\n",
    "                                       stop_codons\n",
    "                                       transcription_start_sites \n",
    "                                    } \n",
    "        \n",
    "        \"\"\"\n",
    "\n",
    "        transcriptome = { \"five_prime_ends\" : [],\n",
    "                    \"three_prime_ends\" : [],\n",
    "                    \"poly_a_sites\" : [],\n",
    "                    \"stop_codons\" :  [],\n",
    "                    \"start_codons\" :  [],\n",
    "                    \"transcription_start_sites\" : []}\n",
    "        \n",
    "        region_and_species = os.path.join(self._regions_dir, self._species)\n",
    "        try:\n",
    "            if flush_cashe:\n",
    "                raise ValueError\n",
    "         \n",
    "            return {region : pybedtools.BedTool(\"%s_%s.bed\" % (region_and_species, \n",
    "                                                               region)) for region in transcriptome}\n",
    "    \n",
    "        except ValueError:\n",
    "            pass\n",
    "\n",
    "        for i, gene_id in enumerate(self._feature_hash.keys()):\n",
    "            if i % 2000 == 0:\n",
    "                print \"processed %d genes\" % (i)\n",
    "                if i == 2000 and limit_genes:\n",
    "                    break\n",
    "                \n",
    "            gene = { \"five_prime_ends\": [],\n",
    "                    \"three_prime_ends\": [],\n",
    "                    \"poly_a_sites\": [],\n",
    "                    \"stop_codons\":  [],\n",
    "                    \"start_codons\":  [],\n",
    "                    \"transcription_start_sites\": []}\n",
    "            try:\n",
    "\n",
    "                for mrna in self._feature_hash[gene_id].keys():\n",
    "\n",
    "                    if mrna == \"gene\":\n",
    "                        continue\n",
    "\n",
    "                    if \"exon\" not in self._feature_hash[gene_id][mrna]:\n",
    "                        continue\n",
    "\n",
    "                    for exon in self._feature_hash[gene_id][mrna]['exon']:\n",
    "                        exon_start = copy.deepcopy(exon)\n",
    "                        exon_start.stop = exon.start + 1\n",
    "\n",
    "\n",
    "                        exon_stop = copy.deepcopy(exon)\n",
    "                        exon_stop.start = exon_stop.stop\n",
    "                        exon_stop.stop += 1\n",
    "\n",
    "                        if exon.strand == \"-\":\n",
    "                            exon_start, exon_stop = exon_stop, exon_start\n",
    "\n",
    "                        if exon_stop.start > exon_stop.stop:\n",
    "                            print exon_stop\n",
    "                            return exon\n",
    "                        gene['five_prime_ends'].append(exon_start)\n",
    "                        gene['three_prime_ends'].append(exon_stop)\n",
    "                \n",
    "                    #transcript vs mRNA need to look at the difference\n",
    "                    if len(self._feature_hash[gene_id][mrna]['transcript']) == 0:\n",
    "                        continue\n",
    "                    elif len(self._feature_hash[gene_id][mrna]['transcript']) > 1:\n",
    "                        print \"error\"\n",
    "                    transcript = self._feature_hash[gene_id][mrna]['transcript'][0]\n",
    "                    transcript_start = copy.deepcopy(transcript)\n",
    "                    transcript_start.stop = transcript.start + 1\n",
    "                    \n",
    "                    transcript_stop = copy.deepcopy(transcript)\n",
    "                    transcript_stop.start = transcript_stop.stop\n",
    "                    transcript_stop.stop += 1\n",
    "                 \n",
    "                    if transcript.strand == \"-\":\n",
    "                        transcript_start, transcript_stop = transcript_stop, transcript_start\n",
    "                        \n",
    "                    gene['poly_a_sites'].append(transcript_stop)\n",
    "                    gene['transcription_start_sites'].append(transcript_start)\n",
    "\n",
    "                    if self._species == \"ce10\": #need to generalize later\n",
    "                        for transcript in self._db.children(gene_id, featuretype=self._feature_names['transcript']):\n",
    "                            try:\n",
    "                                cds = list(self._db.children(transcript,\n",
    "                                                             featuretype='CDS'))\n",
    "\n",
    "                                first_cds, last_cds = cds[0], cds[-1]\n",
    "\n",
    "                                if first_cds.strand == '-':\n",
    "                                    first_cds, last_cds = last_cds, first_cds\n",
    "\n",
    "                                start_codon = first_cds\n",
    "                                start_codon.stop = first_cds.start + 1\n",
    "                                gene['start_codons'].append(start_codon)\n",
    "\n",
    "                                stop_codon = last_cds\n",
    "                                stop_codon.start = stop_codon.stop\n",
    "                                stop_codon.stop  = stop_codon.stop + 1\n",
    "                                gene['stop_codons'].append(stop_codon)\n",
    "\n",
    "                            except:\n",
    "                                pass\n",
    "                    else: #for hg19 and mm9 gencode\n",
    "                        for start_codon in self._feature_hash[gene_id][mrna]['start_codon']:\n",
    "                            start_codon.stop = start_codon.start + 1\n",
    "                            gene['start_codons'].append(start_codon)\n",
    "\n",
    "                        for stop_codon in self._feature_hash[gene_id][mrna]['stop_codon']:\n",
    "                            stop_codon.start = stop_codon.stop\n",
    "                            stop_codon.stop  = stop_codon.stop + 1\n",
    "                            gene['stop_codons'].append(stop_codon)\n",
    "                    \n",
    "            except IndexError:\n",
    "                pass\n",
    "            for region in gene:\n",
    "                transcriptome[region] += self._merge_and_rename_regions(gene[region], gene_id)\n",
    "\n",
    "        for name, intervals in transcriptome.items():\n",
    "            transcriptome[name] = pybedtools.BedTool(map(self._to_bed, intervals)).\\\n",
    "                remove_invalid().sort().each(self._fix_chrom).saveas(\"%s_%s.bed\" % (region_and_species, name))\n",
    "\n",
    "        return transcriptome\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#K562 Expression values\n",
    "foo = urllib.urlretrieve(\"https://www.encodeproject.org/files/ENCFF424CXV/@@download/ENCFF424CXV.tsv\", \"ENCFF424CXV.tsv\")\n",
    "foo = urllib.urlretrieve(\"https://www.encodeproject.org/files/ENCFF073NHK/@@download/ENCFF073NHK.tsv\", \"ENCFF073NHK.tsv\")\n",
    "\n",
    "#HepG2 Expression values\n",
    "foo = urllib.urlretrieve(\"https://www.encodeproject.org/files/ENCFF205WUQ/@@download/ENCFF205WUQ.tsv\", \"ENCFF205WUQ.tsv\")\n",
    "foo = urllib.urlretrieve(\"https://www.encodeproject.org/files/ENCFF915JUZ/@@download/ENCFF915JUZ.tsv\", \"ENCFF915JUZ.tsv\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "k562_rep1 = pd.read_table(\"ENCFF424CXV.tsv\")\n",
    "k562_rep2 = pd.read_table(\"ENCFF073NHK.tsv\")\n",
    "\n",
    "hepg2_rep1 = pd.read_table(\"ENCFF205WUQ.tsv\")\n",
    "hepg2_rep2 = pd.read_table(\"ENCFF915JUZ.tsv\")\n",
    "def get_most_expressed_transcript(transcript_expressions):\n",
    "    transcript_expressions = transcript_expressions[transcript_expressions.TPM > 1]\n",
    "    filtered_tpm = transcript_expressions[transcript_expressions.groupby(\"gene_id\")['TPM'].transform(max) == transcript_expressions['TPM']]\n",
    "    filtered_tpm = filtered_tpm.groupby(\"gene_id\").first()\n",
    "\n",
    "    return filtered_tpm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "k562_highly_expressed_transcripts = set(get_most_expressed_transcript(k562_rep1).transcript_id.values)\n",
    "hepg2_highly_expressed_transcripts = set(get_most_expressed_transcript(hepg2_rep1).transcript_id.values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "db = gffutils.FeatureDB(\"/projects/ps-yeolab/genomes/hg19/gencode/v19/gencode.v19.annotation.gtf.db\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def get_feature_hash(db, expressed_transcripts):\n",
    "    transcripts = db.features_of_type(\"transcript\")\n",
    "    result = defaultdict(lambda: defaultdict(lambda: defaultdict(list)))\n",
    "    for transcript in transcripts:\n",
    "        #Does an isin check\n",
    "        expressed_transcript = (set(transcript.attributes['transcript_id']) & expressed_transcripts)\n",
    "        if len(expressed_transcript) > 0:\n",
    "\n",
    "            #Writes full expressed transcript information\n",
    "            gene = list(db.parents(transcript, featuretype=\"gene\"))\n",
    "            transcript_id = transcript\n",
    "            if len(gene) != 1:\n",
    "                print \"something odd here\"\n",
    "            gene = gene[0]\n",
    "            gene_id = gene.attributes['gene_id'][0]\n",
    "            result[gene_id]['gene'] = gene\n",
    "\n",
    "            if len(expressed_transcript) != 1:\n",
    "                print \"oddness\"\n",
    "\n",
    "            transcript_id = expressed_transcript.pop()\n",
    "\n",
    "            for feature in db.children(transcript):\n",
    "                feature_type = feature.featuretype\n",
    "                result[gene_id][transcript_id][feature_type].append(feature)\n",
    "    return result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File \"/home/gpratt/cell_specific_annotations/hg19_K562_genes.bed\" does not exist\n",
      "processed 0 genes\n",
      "processed 2000 genes\n",
      "processed 4000 genes\n",
      "processed 6000 genes\n",
      "processed 8000 genes\n",
      "processed 10000 genes\n",
      "introns\n",
      "three_prime_utrs\n",
      "proxintron\n",
      "distintron\n",
      "five_prime_utrs\n",
      "cds\n",
      "genes\n",
      "exons\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'cds': <BedTool(/home/gpratt/cell_specific_annotations/hg19_K562_cds.bed)>,\n",
       " 'distintron': <BedTool(/home/gpratt/cell_specific_annotations/hg19_K562_distintron500.bed)>,\n",
       " 'exons': <BedTool(/home/gpratt/cell_specific_annotations/hg19_K562_exons.bed)>,\n",
       " 'five_prime_utrs': <BedTool(/home/gpratt/cell_specific_annotations/hg19_K562_five_prime_utrs.bed)>,\n",
       " 'genes': <BedTool(/home/gpratt/cell_specific_annotations/hg19_K562_genes.bed)>,\n",
       " 'introns': <BedTool(/home/gpratt/cell_specific_annotations/hg19_K562_introns.bed)>,\n",
       " 'proxintron': <BedTool(/home/gpratt/cell_specific_annotations/hg19_K562_proxintron500.bed)>,\n",
       " 'three_prime_utrs': <BedTool(/home/gpratt/cell_specific_annotations/hg19_K562_three_prime_utrs.bed)>}"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "features = GenomicFeatures(\"hg19_K562\", db, regions_dir=\"/home/gpratt/cell_specific_annotations\", gencode=True, expressed_transcripts=k562_highly_expressed_transcripts)\n",
    "features._feature_hash = get_feature_hash(db, k562_highly_expressed_transcripts)\n",
    "\n",
    "#features.get_feature_locations()\n",
    "features.get_genomic_regions()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "processed 0 genes\n",
      "processed 2000 genes\n",
      "processed 4000 genes\n",
      "processed 6000 genes\n",
      "processed 8000 genes\n",
      "processed 10000 genes\n",
      "File \"/home/gpratt/cell_specific_annotations/hg19_HepG2_genes.bed\" does not exist\n",
      "processed 0 genes\n",
      "processed 2000 genes\n",
      "processed 4000 genes\n",
      "processed 6000 genes\n",
      "processed 8000 genes\n",
      "processed 10000 genes\n",
      "introns\n",
      "three_prime_utrs\n",
      "proxintron\n",
      "distintron\n",
      "five_prime_utrs\n",
      "cds\n",
      "genes\n",
      "exons\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'cds': <BedTool(/home/gpratt/cell_specific_annotations/hg19_HepG2_cds.bed)>,\n",
       " 'distintron': <BedTool(/home/gpratt/cell_specific_annotations/hg19_HepG2_distintron500.bed)>,\n",
       " 'exons': <BedTool(/home/gpratt/cell_specific_annotations/hg19_HepG2_exons.bed)>,\n",
       " 'five_prime_utrs': <BedTool(/home/gpratt/cell_specific_annotations/hg19_HepG2_five_prime_utrs.bed)>,\n",
       " 'genes': <BedTool(/home/gpratt/cell_specific_annotations/hg19_HepG2_genes.bed)>,\n",
       " 'introns': <BedTool(/home/gpratt/cell_specific_annotations/hg19_HepG2_introns.bed)>,\n",
       " 'proxintron': <BedTool(/home/gpratt/cell_specific_annotations/hg19_HepG2_proxintron500.bed)>,\n",
       " 'three_prime_utrs': <BedTool(/home/gpratt/cell_specific_annotations/hg19_HepG2_three_prime_utrs.bed)>}"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "features = GenomicFeatures(\"hg19_HepG2\", db, regions_dir=\"/home/gpratt/cell_specific_annotations\", gencode=True, expressed_transcripts=hepg2_highly_expressed_transcripts)\n",
    "features._feature_hash = get_feature_hash(db, hepg2_highly_expressed_transcripts)\n",
    "\n",
    "features.get_feature_locations()\n",
    "features.get_genomic_regions()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
