{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "from collections import Counter\n",
    "import glob\n",
    "from itertools import groupby\n",
    "import os\n",
    "\n",
    "from IPython.core.display import HTML\n",
    "import numpy as np\n",
    "from matplotlib import gridspec\n",
    "import matplotlib.pyplot as mpl\n",
    "import pandas as pd\n",
    "import screed\n",
    "from scipy.stats import ks_2samp\n",
    "import seaborn as sns\n",
    "import pybedtools\n",
    "\n",
    "from clipper.src import CLIP_analysis\n",
    "from clipper.src import kmerdiff\n",
    "from gscripts.general import dataviz\n",
    "\n",
    "img_dir = \"/nas3/gpratt/Dropbox/TAF15/Data/bindnseq/figures\"\n",
    "imp_dir = \"/nas3/gpratt/Dropbox/IMP_team/rbns_figures/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def move_name(interval):\n",
    "    interval.name = interval[12]\n",
    "    return interval\n",
    "\n",
    "def move_name_eric(interval):\n",
    "    interval.name = interval[9]\n",
    "    return interval\n",
    "\n",
    "def trim_intervals(interval):\n",
    "    return pybedtools.interval_constructor(interval[:6])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def calculate_z_score(rbns):\n",
    "    \n",
    "    #get the most enriched column\n",
    "    max_column = rbns.max()\n",
    "    max_column = max_column[max_column == max_column.max()].index[0]\n",
    "    \n",
    "    mean = rbns[max_column].mean()\n",
    "    std  = rbns[max_column].std()\n",
    "\n",
    "    zscore = rbns[max_column].apply(lambda x: (x - mean) / std)\n",
    "    return zscore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2017-03-15 16:54:04--  https://www.encodeproject.org/files/ENCFF002DFF/@@download/ENCFF002DFF.tsv\n",
      "Resolving www.encodeproject.org... 171.67.205.70\n",
      "Connecting to www.encodeproject.org|171.67.205.70|:443... connected.\n",
      "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
      "Location: https://download.encodeproject.org/http://encode-files.s3.amazonaws.com/2014/06/10/183cd97a-ee70-4145-9111-3da9bb1f8395/ENCFF002DFF.tsv?Signature=syyw5a5FFCvVKcfR%2FxhBupl3rJw%3D&Expires=1489751644&AWSAccessKeyId=ASIAI64XNBX5DTOV6DSQ&response-content-disposition=attachment%3B%20filename%3DENCFF002DFF.tsv&x-amz-security-token=FQoDYXdzEOH//////////wEaDIeA%2Bv%2BlE8LoUNlhQyK3A2lj8AMCV6LR7UGb7NuciZfMW5KqgNWJDfaNFNqp74yTCASfH6HbQAmmwEawEY1rWt0kNsBgo1ZBU0KVGCPREsWsn8kWp7x2dNgcXKpmsj8PlNpTfJCKopgZ4ZHTZaMhS3WQ1bqkGeT1T41t27VGHMi91W42VgGNg%2B3iFrFdV9gywd79pAWugx9eb8vH8AI6rJY8uV3ZDhPLinL2GoUm5hvIeT4%2BiksN4HVWKDBpVKQwBtAh3HKcgkM3YVYoPQTXzQpzVIUnXF2QhUzN2%2B4GzGU%2BQG%2Bkfn7TY//XZIyZUtYviae0PMoQ1ktFmAtosbqzr8QhEupNEy/fQwUkeAfMj6UmxwK0c3cryFb4FLj4DGAbn2st9ZJ%2B0huV9CEcmKGJtt96vxMOo5Eco6kweuaweINRS83oskWATja9vYV7PltONb/SAUkmEUpsKP%2BxJYyzx52e8AVOgz%2BZkHrPieZKfYj5CYfwnOa79ILMlJkPMgnRgoIc7B5FoQ1F0jCaFQQUXEEcasRXvpzz/9VBS4iK3J1VB0qjzmTOMcADR1OYrX7uAjmYxV/uUKOHapU4BG5CFs4L2npRGhUo26WnxgU%3D [following]\n",
      "--2017-03-15 16:54:04--  https://download.encodeproject.org/http://encode-files.s3.amazonaws.com/2014/06/10/183cd97a-ee70-4145-9111-3da9bb1f8395/ENCFF002DFF.tsv?Signature=syyw5a5FFCvVKcfR%2FxhBupl3rJw%3D&Expires=1489751644&AWSAccessKeyId=ASIAI64XNBX5DTOV6DSQ&response-content-disposition=attachment%3B%20filename%3DENCFF002DFF.tsv&x-amz-security-token=FQoDYXdzEOH//////////wEaDIeA%2Bv%2BlE8LoUNlhQyK3A2lj8AMCV6LR7UGb7NuciZfMW5KqgNWJDfaNFNqp74yTCASfH6HbQAmmwEawEY1rWt0kNsBgo1ZBU0KVGCPREsWsn8kWp7x2dNgcXKpmsj8PlNpTfJCKopgZ4ZHTZaMhS3WQ1bqkGeT1T41t27VGHMi91W42VgGNg%2B3iFrFdV9gywd79pAWugx9eb8vH8AI6rJY8uV3ZDhPLinL2GoUm5hvIeT4%2BiksN4HVWKDBpVKQwBtAh3HKcgkM3YVYoPQTXzQpzVIUnXF2QhUzN2%2B4GzGU%2BQG%2Bkfn7TY//XZIyZUtYviae0PMoQ1ktFmAtosbqzr8QhEupNEy/fQwUkeAfMj6UmxwK0c3cryFb4FLj4DGAbn2st9ZJ%2B0huV9CEcmKGJtt96vxMOo5Eco6kweuaweINRS83oskWATja9vYV7PltONb/SAUkmEUpsKP%2BxJYyzx52e8AVOgz%2BZkHrPieZKfYj5CYfwnOa79ILMlJkPMgnRgoIc7B5FoQ1F0jCaFQQUXEEcasRXvpzz/9VBS4iK3J1VB0qjzmTOMcADR1OYrX7uAjmYxV/uUKOHapU4BG5CFs4L2npRGhUo26WnxgU%3D\n",
      "Resolving download.encodeproject.org... 171.67.205.70\n",
      "Connecting to download.encodeproject.org|171.67.205.70|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 397360 (388K) [text/tab-separated-values]\n",
      "Saving to: `ENCFF002DFF.tsv.1'\n",
      "\n",
      "100%[======================================>] 397,360     --.-K/s   in 0.1s    \n",
      "\n",
      "2017-03-15 16:54:04 (2.73 MB/s) - `ENCFF002DFF.tsv.1' saved [397360/397360]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://www.encodeproject.org/files/ENCFF002DFF/@@download/ENCFF002DFF.tsv "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "rbfox2 = pd.read_table(\"ENCFF002DFF.tsv\", index_col=0)\n",
    "rbfox2_zscore = calculate_z_score(rbfox2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "peaks_dir = \"/home/gpratt/projects/encode/analysis/peak_reanalysis_v14\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def make_clipper_ish(interval):\n",
    "    interval.name = interval[7]\n",
    "    interval[6] = interval.start\n",
    "    interval[7] = interval.stop\n",
    "    return interval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "submitted_datasets = pd.read_table(\"/home/elvannostrand/data/clip/CLIPseq_analysis/ENCODE_CLIPperv2_20170130/ALLDATASETS_submittedonly.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/projects/ps-yeolab3/encode/analysis/encode_v12/271_01_HNRNPC.merged.r2.bam'"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "submitted_datasets.CLIP_rep1[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "glob.glob(os.path.join(peaks_dir, \"*.clipper.bed.clip_formatted.bed\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "regions, all_regions = CLIP_analysis.regions_generator()\n",
    "\n",
    "genes = pybedtools.BedTool(\"/home/gpratt/clipper/clipper/data/regions/hg19_genes.bed\")\n",
    "\n",
    "rbps = \"\"\"204_01.basedon_204_01.peaks.l2inputnormnew.bed.compressed.bed.annotatedl2fc_3_pval_3.clipper.bed.clip_formatted.bed\n",
    "204_02.basedon_204_02.peaks.l2inputnormnew.bed.compressed.bed.annotatedl2fc_3_pval_3.clipper.bed.clip_formatted.bed\n",
    "676_01.basedon_676_01.peaks.l2inputnormnew.bed.compressed.bed.annotatedl2fc_3_pval_3.clipper.bed.clip_formatted.bed\n",
    "676_02.basedon_676_02.peaks.l2inputnormnew.bed.compressed.bed.annotatedl2fc_3_pval_3.clipper.bed.clip_formatted.bed\"\"\".split()\n",
    "\n",
    "rbps = {rbp: pybedtools.BedTool(os.path.join(peaks_dir, rbp)) for rbp in rbps}\n",
    "\n",
    "reprocessed_rbps = {}\n",
    "for name, rbp in rbps.items():\n",
    "    #overlapping_genes = rbp.intersect(genes, wo=True, s=True).each(tmp_move_name).each(trim_intervals).saveas()\n",
    "\n",
    "    reprocessed_rbps[name] = CLIP_analysis.assign_to_regions(rbp, os.path.splitext(rbp.fn)[0], \n",
    "                                    assigned_dir=\"assigned\", nrand=10, species=\"hg19\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def kmer_enrichment(header):\n",
    "    fasta_files = glob.glob(\"fasta/{}.*.fa\".format(header))\n",
    "    kmers = {}\n",
    "    for group, lst in groupby(sorted(fasta_files), key=lambda x: os.path.basename(x).split(\".\")[-3]):\n",
    "        lst = list(lst)\n",
    "        if len(lst) != 2:\n",
    "            print group\n",
    "            print \"bad length, continuing\", group\n",
    "            continue\n",
    "\n",
    "        random, real = lst\n",
    "        kmer = kmerdiff.kmer_diff(real, random, 6)\n",
    "        kmer = pd.DataFrame(kmer[0]).T\n",
    "        kmer.columns = ['freq1', 'freq2', 'delta']\n",
    "        kmer.index.name = \"kmer\"\n",
    "        kmers[group] = kmer\n",
    "    return pd.concat(kmers)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "for name in reprocessed_rbps:\n",
    "    CLIP_analysis.make_fasta_files_from_regions(reprocessed_rbps[name], name, \"fasta\", \"/projects/ps-yeolab/genomes/hg19/chromosomes/all.fa\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "reprocessed_kmer_dict = {}\n",
    "for name in reprocessed_rbps:\n",
    "    reprocessed_kmer_dict[name] = kmer_enrichment(name)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def plot_bindnseq(zscore, kmer, img_path, motifs=[\"GGTA\"], name=\"\", colors=None, ylim=None):\n",
    "    num_rows = 4\n",
    "    num_cols = 4\n",
    "    \n",
    "    if type(motifs) == str:\n",
    "        color = ['r']\n",
    "        motifs = [motifs]\n",
    "        \n",
    "    with dataviz.Figure(img_path, figsize=(3.0, 3.0)) as fig:\n",
    "        gs = gridspec.GridSpec(num_rows, num_cols)\n",
    "\n",
    "        joined = pd.concat([kmer.delta, zscore], axis=1)\n",
    "        joined.columns = ['kmer', 'rbns']\n",
    "        \n",
    "        ax = fig.add_subplot(gs[0, 0:3])\n",
    "\n",
    "        sns.kdeplot(joined['rbns'], ax=ax, color='k')\n",
    "        for motif, color in zip(motifs, colors):\n",
    "            filtered = joined[[motif in item for item in joined.index]]\n",
    "\n",
    "            sns.kdeplot(filtered['rbns'], ax=ax, color=color)\n",
    "            ax.text(.8, 1, \"p-val {:.2}\".format(ks_2samp(joined['rbns'], filtered['rbns'])[1]),\n",
    "                    transform=ax.transAxes, fontsize=8, color=color)\n",
    "\n",
    "        legend = ax.legend()\n",
    "        legend.remove()\n",
    "        ax.set_yticks([])\n",
    "        ax.set_xticks([])\n",
    "        sns.despine(ax=ax, left=True)\n",
    "        [tick.set_fontsize(8) for tick in ax.get_xticklabels()]\n",
    "\n",
    "        ax = fig.add_subplot(gs[1:4, 3])\n",
    "        sns.kdeplot(joined['kmer'], ax=ax, vertical=True, color='k')\n",
    "        for motif, color in zip(motifs, colors):\n",
    "            filtered = joined[[motif in item for item in joined.index]]\n",
    "            sns.kdeplot(filtered['kmer'], ax=ax, vertical=True, color=color)\n",
    "            ax.text(.1,.05, \"p-val {:.2}\".format(ks_2samp(joined['kmer'], filtered['kmer'])[1]),\n",
    "                    transform=ax.transAxes, fontsize=8, color=color)\n",
    "\n",
    "        ax.set_xticks([])\n",
    "        ax.set_yticks([])\n",
    "        legend = ax.legend()\n",
    "        legend.remove()\n",
    "        sns.despine(ax=ax, bottom=True)\n",
    "        [tick.set_fontsize(8) for tick in ax.get_xticklabels()]\n",
    "        if ylim is not None:\n",
    "            ax.set_ylim((min(kmer.delta), ylim))\n",
    "            \n",
    "        ax = fig.add_subplot(gs[1:4,0:3])\n",
    "\n",
    "        found_motifs = set([])\n",
    "        for motif, color in zip(motifs, colors):\n",
    "            pass\n",
    "            filtered = joined[[motif in item for item in joined.index]]\n",
    "            line2 = ax.scatter(filtered['rbns'], \n",
    "                               filtered['kmer'],\n",
    "                               #alpha=.5, \n",
    "                               s=5, \n",
    "                               label=\"Contains {}\".format(motif),\n",
    "                               color=color,\n",
    "                               linewidths=0,\n",
    "                               #rasterized=True\n",
    "                          )\n",
    "            found_motifs = found_motifs.union(set(filtered.index))\n",
    "        diff = joined.index.difference(found_motifs)\n",
    "        \n",
    "        other_motifs = joined.ix[diff]\n",
    "        line1 = ax.scatter(other_motifs['rbns'], \n",
    "                           other_motifs['kmer'], \n",
    "                           #alpha=.5,\n",
    "                           color='.01', \n",
    "                           s=1, \n",
    "                           label=\"Other\", \n",
    "                           linewidths=0,\n",
    "                           #rasterized=True\n",
    "                          )\n",
    "        \n",
    "        if ylim is not None:\n",
    "            ax.set_ylim((min(kmer.delta), ylim))\n",
    "        ax.set_ylabel(name + \"\\nCLIP Cluster 6-mer Enrichment\")\n",
    "        lgnd = ax.legend(loc=4, scatterpoints=1, fontsize=8)\n",
    "        #ax.set_xticks([])\n",
    "        ax.set_xlabel(name + \" RBNS R Z-Score\")\n",
    "\n",
    "        sns.despine(ax=ax)\n",
    "        lgnd.legendHandles[0]._sizes = [30]\n",
    "        lgnd.legendHandles[1]._sizes = [30]\n",
    "        print ax.get_ylim()\n",
    "\n",
    "def save_all_plots(kmer_dict, zscore, fn, name=\"\", motifs=\"CACA\", colors='r', img_dir=\"/nas3/gpratt/Dropbox/TAF15/Data/bindnseq/figures\"):\n",
    "    for region in kmer_dict.index.levels[0]:\n",
    "        plot_bindnseq(zscore, kmer_dict.ix[region], os.path.join(img_dir,'{}_{}.png'.format(region, fn)), motifs=motifs, name=name, colors=colors )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "img_dir = \"/projects/ps-yeolab3/encode/analysis/website_figures/\"\n",
    "for name, kmers in reprocessed_kmer_dict.items():\n",
    "    output_name = name.split(\".\")[0]\n",
    "    save_all_plots(kmers, \n",
    "               rbfox2_zscore,\n",
    "               name,\n",
    "               output_name,\n",
    "               motifs=['TGCA', 'GCAT', 'CATG'], \n",
    "               colors=['y', '#EB2526', '#3856A1'], \n",
    "               img_dir=img_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Do the kmer thing using the data I've already generated..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gpratt/anaconda/lib/python2.7/site-packages/matplotlib/__init__.py:1350: UserWarning:  This call to matplotlib.use() has no effect\n",
      "because the backend has already been chosen;\n",
      "matplotlib.use() must be called *before* pylab, matplotlib.pyplot,\n",
      "or matplotlib.backends is imported for the first time.\n",
      "\n",
      "  warnings.warn(_use_error_msg)\n"
     ]
    }
   ],
   "source": [
    "from clipper.src import CLIP_analysis_display"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "out_dir = \"/projects/ps-yeolab3/encode/analysis/kmer_freqs\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_pickle_file(uID, rep):\n",
    "    rbp = \"/home/gpratt/projects/encode/analysis/peak_reanalysis_v14/{uid}_{rep}.basedon_{uid}_{rep}.peaks.l2inputnormnew.bed.compressed.bed.annotatedl2fc_3_pval_3.clipper.bed.clip_formatted.bed.clip_analysis.pickle\".format(uid=uID, rep=rep)\n",
    "    viz = CLIP_analysis_display.ClipVisualization(open(rbp))\n",
    "    return viz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "249\n",
      "267\n",
      "275\n",
      "281\n",
      "283\n",
      "289\n",
      "339\n",
      "353\n",
      "417\n",
      "452\n",
      "484\n",
      "537\n",
      "553\n",
      "522\n",
      "575\n"
     ]
    }
   ],
   "source": [
    "for name, row in submitted_datasets.iterrows():\n",
    "    try:\n",
    "        pickle_file = get_pickle_file(row.uID, \"01\")\n",
    "\n",
    "        results = {}\n",
    "        for region in pickle_file.kmer_results:\n",
    "            results[region] = pd.DataFrame(pickle_file.kmer_results[region][6], index=['freq1', 'freq2', 'delta'])\n",
    "        results = pd.concat(results).T\n",
    "        results.index.name = \"kmer\"\n",
    "        results.columns = [\"_\".join(column) for column in results.columns]\n",
    "        results.to_csv(os.path.join(out_dir, \"{}_{}_{}_{}.csv\".format(row.uID, row.RBP, row['Cell line'], \"01\")))\n",
    "\n",
    "        pickle_file = get_pickle_file(row.uID, \"02\")\n",
    "\n",
    "        results = {}\n",
    "        for region in pickle_file.kmer_results:\n",
    "            results[region] = pd.DataFrame(pickle_file.kmer_results[region][6], index=['freq1', 'freq2', 'delta'])\n",
    "        results = pd.concat(results).T\n",
    "        results.index.name = \"kmer\"\n",
    "        results.columns = [\"_\".join(column) for column in results.columns]\n",
    "        results.to_csv(os.path.join(out_dir, \"{}_{}_{}_{}.csv\".format(row.uID, row.RBP, row['Cell line'], \"02\")))\n",
    "    except:\n",
    "        print row.uID\n",
    "        continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
