{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import subprocess\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from scipy.stats import hypergeom\n",
    "from IPython.core.display import HTML\n",
    "import pybedtools\n",
    "import os\n",
    "import glob\n",
    "from IPython.core.display import HTML\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def annotatePeaks_stranded(peaks, motif, species=\"hg19\", hist=10, size=1000):\n",
    "    \n",
    "    \"\"\"\n",
    "    \n",
    "    runs homer with standard args\n",
    "    output location is saved\n",
    "    \n",
    "    foreground - str, location of fasta file for the foreground distribution\n",
    "    background - str, location of fasta file for the background distribution\n",
    "    k - different k-mers to examine\n",
    "    outloc - directory to output homer results \n",
    "\n",
    "    --make optional make work off locations and not fasta files \n",
    "    \n",
    "    \"\"\"\n",
    "    #findMotifs.pl clusters.fa fasta outloc -nofacts p 4 -rna -S 10 -len 5,6,7,8,9 -noconvert -nogo -fasta background.fa\n",
    "    #converts k to a string for use in subprocess\n",
    "    #I'm forgetting strand specific information here\n",
    "    try:\n",
    "        with open(\"tmp.txt\", 'w') as out:\n",
    "            subprocess.call([\"annotatePeaks.pl\",\n",
    "                             peaks,\n",
    "                             species,\n",
    "                             \"-m\", motif,\n",
    "                             #\"-hist\", str(hist),\n",
    "                             \"-size\", \"given\",\n",
    "                             \"-noann\"], shell=False, stdout=out)\n",
    "    except OSError:\n",
    "        print \"Homer not installed, ignoring motif generation, install homer for this to work\"  \n",
    "        raise   \n",
    "    annotation = pd.read_table(\"tmp.txt\", index_col=0)\n",
    "    annotation['stranded_peaks'] = annotation.apply(only_stranded_motifs, axis=1)\n",
    "    return annotation\n",
    "\n",
    "def only_stranded_motifs(row):\n",
    "    motifs = row[-1]\n",
    "    try:\n",
    "        true_peaks = \"),\".join([motif for motif in motifs.split(\"),\") if row.Strand in motif])\n",
    "        if len(true_peaks) == 0:\n",
    "            return np.nan\n",
    "        \n",
    "        return true_peaks\n",
    "    except AttributeError:\n",
    "        return motifs\n",
    "    \n",
    "def get_strand_of_motif(motifs):\n",
    "    try:\n",
    "        return \",\".join(list(set(motif.split(\",\")[1] for motif in motifs.split(\"),\"))))\n",
    "    except AttributeError:\n",
    "        return np.nan\n",
    "    \n",
    "def bound(row):\n",
    "    try:\n",
    "        return row.Strand in row.motif_strands\n",
    "    except TypeError:\n",
    "        return False\n",
    "    \n",
    "def get_bound_motifs(bedtool, motif_file, motif_name):\n",
    "    annoted_peaks = annotatePeaks(bedtool.fn, motif_file, \"mm9\")\n",
    "    annoted_peaks['motif_strands'] = annoted_peaks['1-{} Distance From Peak(sequence,strand,conservation)'.format(motif_name)].apply(get_strand_of_motif)\n",
    "    annoted_peaks['motif_present'] = annoted_peaks.apply(bound, axis=1)\n",
    "    return annoted_peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def findMotifsGenome(peaks, motif, species=\"hg19\"):\n",
    "    try:\n",
    "        with open(\"tmp.txt\", 'w') as out:\n",
    "            subprocess.call([\"findMotifsGenome.pl\",\n",
    "                             peaks,\n",
    "                             species,\n",
    "                             \"foo\",\n",
    "                             \"-size\", \"given \",\n",
    "                             \"-rna\",\n",
    "                             \"-find\", motif,\n",
    "                            ], shell=False, stdout=out)\n",
    "    except OSError:\n",
    "        print \"Homer not installed, ignoring motif generation, install homer for this to work\"  \n",
    "        raise   \n",
    "    return pd.read_table(\"tmp.txt\", index_col=0)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def findMotifsGenome(peaks, motif, species=\"hg19\"):\n",
    "    !findMotifsGenome.pl $peaks $species foo -size given -rna -find $motif > tmp.txt\n",
    "    return pd.read_table(\"tmp.txt\", index_col=0)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "rbfox2_motif = \"/home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#dir_name = \"/projects/ps-yeolab2/encode/analysis/encode_v9/assigned/\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# v12 = glob.glob(\"/projects/ps-yeolab2/encode/analysis/encode_v12/assigned/*\")\n",
    "# v8 = [item.replace(\"/projects/ps-yeolab2/encode/analysis/encode_v12/assigned/\", \n",
    "#                    \"/home/gpratt/projects/encode/analysis/encode_v8/assigned/\") for item in v12]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# v12_peak_sizes = {}\n",
    "# for item in v12:\n",
    "#     try:\n",
    "#         if \"all\" in item and \"real\" in item:\n",
    "#             v12_peak_sizes[os.path.basename(item)] = len(pybedtools.BedTool(item)) \n",
    "#     except:\n",
    "#         pass\n",
    "    \n",
    "# v8_peak_sizes = {}\n",
    "# for item in v8:\n",
    "#     try:\n",
    "#         if \"all\" in item and \"real\" in item:\n",
    "#             v8_peak_sizes[os.path.basename(item)] = len(pybedtools.BedTool(item))\n",
    "#     except:\n",
    "#         pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# foo = pd.concat({\"v12\": pd.Series(v12_peak_sizes),\n",
    "#                  \"v8\": pd.Series(v8_peak_sizes)}).unstack().T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# foo = foo[[\"all\" in item for item in foo.index]]\n",
    "# foo = foo[[\"real\" in item for item in foo.index]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# foo['same'] = foo.v12 == foo.v8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# HTML(foo.to_html())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# for item in foo.index:\n",
    "#     !diff /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/$item /home/gpratt/projects/encode/analysis/encode_v8/assigned/$item"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "dir_name = \"/home/gpratt/projects/encode/analysis/encode_v8/assigned\"\n",
    "peaks_to_annotate = {\"RBFOX2_293XT_1to200_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"RBFOX2_293XT_1to100_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"RBFOX2_293XT_1to50_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"RBFOX2_293XT_1to25_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"RBFOX2_293XT_1to12_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP10_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP8_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP1_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP5_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP6_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),}\n",
    "\n",
    "dir_name = \"/projects/ps-yeolab2/encode/analysis/encode_v12/assigned/\"\n",
    "peaks_to_annotate_v12 = {\"RBFOX2_293XT_1to200_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"RBFOX2_293XT_1to100_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"RBFOX2_293XT_1to50_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"RBFOX2_293XT_1to25_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"RBFOX2_293XT_1to12_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP10_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP8_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP1_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP5_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "                     \"EVN_041315_CLIP6_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),}\n",
    "\n",
    "\n",
    "# dir_name = \"/projects/ps-yeolab2/encode/analysis/encode_v12/assigned/\"\n",
    "# peaks_to_annotate = {\"RBFOX2_293XT_1to200_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "#                      \"RBFOX2_293XT_1to100_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "#                      \"RBFOX2_293XT_1to50_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "#                      \"RBFOX2_293XT_1to25_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "#                      \"RBFOX2_293XT_1to12_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "#                      \"EVN_041315_CLIP10_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "#                      \"EVN_041315_CLIP8_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "#                      \"EVN_041315_CLIP1_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "#                      \"EVN_041315_CLIP5_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),\n",
    "#                      \"EVN_041315_CLIP6_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed.all.real.BED\"),}\n",
    "\n",
    "\n",
    "peaks_to_rand_1_annotate = {\"RBFOX2_293XT_1to200_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),\n",
    "                     \"RBFOX2_293XT_1to100_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),\n",
    "                     \"RBFOX2_293XT_1to50_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),\n",
    "                     \"RBFOX2_293XT_1to25_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),\n",
    "                     \"RBFOX2_293XT_1to12_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),\n",
    "                     \"EVN_041315_CLIP10_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),\n",
    "                     \"EVN_041315_CLIP8_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),\n",
    "                     \"EVN_041315_CLIP1_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),\n",
    "                     \"EVN_041315_CLIP5_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),\n",
    "                     \"EVN_041315_CLIP6_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\"),}\n",
    "\n",
    "peaks_to_rand_2_annotate = {\"RBFOX2_293XT_1to200_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),\n",
    "                     \"RBFOX2_293XT_1to100_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),\n",
    "                     \"RBFOX2_293XT_1to50_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),\n",
    "                     \"RBFOX2_293XT_1to25_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),\n",
    "                     \"RBFOX2_293XT_1to12_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),\n",
    "                     \"EVN_041315_CLIP10_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),\n",
    "                     \"EVN_041315_CLIP8_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),\n",
    "                     \"EVN_041315_CLIP1_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),\n",
    "                     \"EVN_041315_CLIP5_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),\n",
    "                     \"EVN_041315_CLIP6_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\"),}\n",
    "\n",
    "peaks_to_rand_3_annotate = {\"RBFOX2_293XT_1to200_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),\n",
    "                     \"RBFOX2_293XT_1to100_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),\n",
    "                     \"RBFOX2_293XT_1to50_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),\n",
    "                     \"RBFOX2_293XT_1to25_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),\n",
    "                     \"RBFOX2_293XT_1to12_gel_RBFOX2\": os.path.join(dir_name, \"RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),\n",
    "                     \"EVN_041315_CLIP10_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),\n",
    "                     \"EVN_041315_CLIP8_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),\n",
    "                     \"EVN_041315_CLIP1_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),\n",
    "                     \"EVN_041315_CLIP5_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),\n",
    "                     \"EVN_041315_CLIP6_RBFOX2\": os.path.join(dir_name, \"EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\"),}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed.all.real.BED EVN_041315_CLIP10_RBFOX2 33194\n",
      "/projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed.all.real.BED EVN_041315_CLIP1_RBFOX2 27038\n",
      "/projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed.all.real.BED EVN_041315_CLIP5_RBFOX2 20718\n",
      "/projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed.all.real.BED EVN_041315_CLIP6_RBFOX2 914\n",
      "/projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed.all.real.BED EVN_041315_CLIP8_RBFOX2 20495\n",
      "/projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED RBFOX2_293XT_1to100_gel_RBFOX2 34934\n",
      "/projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED RBFOX2_293XT_1to12_gel_RBFOX2 33665\n",
      "/projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED RBFOX2_293XT_1to200_gel_RBFOX2 22370\n",
      "/projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED RBFOX2_293XT_1to25_gel_RBFOX2 27088\n",
      "/projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED RBFOX2_293XT_1to50_gel_RBFOX2 28458\n"
     ]
    }
   ],
   "source": [
    "for name, fn in sorted(peaks_to_annotate_v12.items(), key=lambda x: x[0]):\n",
    "    print fn, name, len(pybedtools.BedTool(fn))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/gpratt/projects/encode/analysis/encode_v8/assigned/EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed.all.real.BED EVN_041315_CLIP10_RBFOX2 33194\n",
      "/home/gpratt/projects/encode/analysis/encode_v8/assigned/EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed.all.real.BED EVN_041315_CLIP1_RBFOX2 27038\n",
      "/home/gpratt/projects/encode/analysis/encode_v8/assigned/EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed.all.real.BED EVN_041315_CLIP5_RBFOX2 20718\n",
      "/home/gpratt/projects/encode/analysis/encode_v8/assigned/EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed.all.real.BED EVN_041315_CLIP6_RBFOX2 914\n",
      "/home/gpratt/projects/encode/analysis/encode_v8/assigned/EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed.all.real.BED EVN_041315_CLIP8_RBFOX2 20495\n",
      "/home/gpratt/projects/encode/analysis/encode_v8/assigned/RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED RBFOX2_293XT_1to100_gel_RBFOX2 34934\n",
      "/home/gpratt/projects/encode/analysis/encode_v8/assigned/RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED RBFOX2_293XT_1to12_gel_RBFOX2 33665\n",
      "/home/gpratt/projects/encode/analysis/encode_v8/assigned/RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED RBFOX2_293XT_1to200_gel_RBFOX2 22370\n",
      "/home/gpratt/projects/encode/analysis/encode_v8/assigned/RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED RBFOX2_293XT_1to25_gel_RBFOX2 27088\n",
      "/home/gpratt/projects/encode/analysis/encode_v8/assigned/RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED RBFOX2_293XT_1to50_gel_RBFOX2 28458\n"
     ]
    }
   ],
   "source": [
    "for name, fn in sorted(peaks_to_annotate_v8.items(), key=lambda x: x[0]):\n",
    "    try:\n",
    "        \n",
    "        print fn, name, len(pybedtools.BedTool(fn))\n",
    "    except:\n",
    "        pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def bound_in_same_strand(row):\n",
    "    try:\n",
    "        motifs = row['1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_0.6(0.999) Distance From Peak(sequence,strand,conservation)'].split(\"),\")\n",
    "        return any(row.Strand in item.split(\",\")[1] for item in motifs)\n",
    "    except:\n",
    "        return False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#This approach is incorrect, doesn't get strand quite right and isn't as stringent with the kmer finding as the approach below\n",
    "# results_annotate = {}\n",
    "# for peak in peaks_to_annotate:\n",
    "#     annotated_peak = annotatePeaks_stranded(peaks_to_annotate[peak], rbfox2_motif)\n",
    "#     results_annotate[peak] = {\"peaks_with_motif\": len(annotated_peak.stranded_peaks.dropna()), \"num_peaks\": len(pybedtools.BedTool(peaks_to_annotate[peak]))}\n",
    "    \n",
    "# results_rand_1_annotate = {}\n",
    "# for peak in peaks_to_rand_1_annotate:\n",
    "#     annotated_peak = annotatePeaks_stranded(peaks_to_rand_1_annotate[peak], rbfox2_motif)\n",
    "#     #annotated_peak['bound_in_same_strand'] = annotated_peak.apply(bound_in_same_strand, axis=1)\n",
    "#     results_rand_1_annotate[peak] = {\"peaks_with_motif\": len(annotated_peak.stranded_peaks.dropna()), \"num_peaks\": len(pybedtools.BedTool(peaks_to_annotate[peak]))}\n",
    "    \n",
    "# results_rand_2_annotate = {}\n",
    "# for peak in peaks_to_rand_2_annotate:\n",
    "#     annotated_peak = annotatePeaks_stranded(peaks_to_rand_2_annotate[peak], rbfox2_motif)\n",
    "#     annotated_peak['bound_in_same_strand'] = annotated_peak.apply(bound_in_same_strand, axis=1)\n",
    "#     results_rand_2_annotate[peak] = {\"peaks_with_motif\": len(annotated_peak.stranded_peaks.dropna()), \"num_peaks\": len(pybedtools.BedTool(peaks_to_annotate[peak]))}\n",
    "    \n",
    "# results_rand_3_annotate = {}\n",
    "# for peak in peaks_to_rand_3_annotate:\n",
    "#     annotated_peak = annotatePeaks_stranded(peaks_to_rand_3_annotate[peak], rbfox2_motif)\n",
    "#     #annotated_peak['bound_in_same_strand'] = annotated_peak.apply(bound_in_same_strand, axis=1)\n",
    "#     results_rand_3_annotate[peak] = {\"peaks_with_motif\": len(annotated_peak.stranded_peaks.dropna()), \"num_peaks\": len(pybedtools.BedTool(peaks_to_annotate[peak]))}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 33194\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 33194\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 38 (avg size of targets)\n",
      "\tBackground files for 38 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 3573 sequences from chr1\n",
      "\tExtracting 2082 sequences from chr2\n",
      "\tExtracting 1760 sequences from chr3\n",
      "\tExtracting 892 sequences from chr4\n",
      "\tExtracting 1434 sequences from chr5\n",
      "\tExtracting 1597 sequences from chr6\n",
      "\tExtracting 1560 sequences from chr7\n",
      "\tExtracting 1000 sequences from chr8\n",
      "\tExtracting 1209 sequences from chr9\n",
      "\tExtracting 1113 sequences from chr10\n",
      "\tExtracting 1863 sequences from chr11\n",
      "\tExtracting 1975 sequences from chr12\n",
      "\tExtracting 427 sequences from chr13\n",
      "\tExtracting 1173 sequences from chr14\n",
      "\tExtracting 1041 sequences from chr15\n",
      "\tExtracting 1874 sequences from chr16\n",
      "\tExtracting 2592 sequences from chr17\n",
      "\tExtracting 414 sequences from chr18\n",
      "\tExtracting 2207 sequences from chr19\n",
      "\tExtracting 786 sequences from chr20\n",
      "\tExtracting 412 sequences from chr21\n",
      "\tExtracting 1038 sequences from chr22\n",
      "\tExtracting 1104 sequences from chrX\n",
      "\tExtracting 68 sequences from chrM\n",
      "\t511 of 33194 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t32683 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 914\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 914\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 35 (avg size of targets)\n",
      "\tBackground files for 35 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 77 sequences from chr1\n",
      "\tExtracting 47 sequences from chr2\n",
      "\tExtracting 35 sequences from chr3\n",
      "\tExtracting 33 sequences from chr4\n",
      "\tExtracting 28 sequences from chr5\n",
      "\tExtracting 30 sequences from chr6\n",
      "\tExtracting 57 sequences from chr7\n",
      "\tExtracting 35 sequences from chr8\n",
      "\tExtracting 42 sequences from chr9\n",
      "\tExtracting 53 sequences from chr10\n",
      "\tExtracting 48 sequences from chr11\n",
      "\tExtracting 33 sequences from chr12\n",
      "\tExtracting 12 sequences from chr13\n",
      "\tExtracting 27 sequences from chr14\n",
      "\tExtracting 13 sequences from chr15\n",
      "\tExtracting 59 sequences from chr16\n",
      "\tExtracting 53 sequences from chr17\n",
      "\tExtracting 11 sequences from chr18\n",
      "\tExtracting 65 sequences from chr19\n",
      "\tExtracting 40 sequences from chr20\n",
      "\tExtracting 17 sequences from chr21\n",
      "\tExtracting 40 sequences from chr22\n",
      "\tExtracting 20 sequences from chrX\n",
      "\tExtracting 39 sequences from chrM\n",
      "\t1 of 914 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t913 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 20495\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 20495\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 33 (avg size of targets)\n",
      "\tBackground files for 33 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2191 sequences from chr1\n",
      "\tExtracting 1245 sequences from chr2\n",
      "\tExtracting 1150 sequences from chr3\n",
      "\tExtracting 560 sequences from chr4\n",
      "\tExtracting 800 sequences from chr5\n",
      "\tExtracting 945 sequences from chr6\n",
      "\tExtracting 885 sequences from chr7\n",
      "\tExtracting 564 sequences from chr8\n",
      "\tExtracting 761 sequences from chr9\n",
      "\tExtracting 685 sequences from chr10\n",
      "\tExtracting 1153 sequences from chr11\n",
      "\tExtracting 1170 sequences from chr12\n",
      "\tExtracting 249 sequences from chr13\n",
      "\tExtracting 597 sequences from chr14\n",
      "\tExtracting 581 sequences from chr15\n",
      "\tExtracting 1208 sequences from chr16\n",
      "\tExtracting 1652 sequences from chr17\n",
      "\tExtracting 211 sequences from chr18\n",
      "\tExtracting 1489 sequences from chr19\n",
      "\tExtracting 541 sequences from chr20\n",
      "\tExtracting 279 sequences from chr21\n",
      "\tExtracting 757 sequences from chr22\n",
      "\tExtracting 759 sequences from chrX\n",
      "\tExtracting 63 sequences from chrM\n",
      "\t277 of 20495 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t20218 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 22370\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 22370\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 36 (avg size of targets)\n",
      "\tBackground files for 36 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2386 sequences from chr1\n",
      "\tExtracting 1238 sequences from chr2\n",
      "\tExtracting 1147 sequences from chr3\n",
      "\tExtracting 544 sequences from chr4\n",
      "\tExtracting 807 sequences from chr5\n",
      "\tExtracting 958 sequences from chr6\n",
      "\tExtracting 1023 sequences from chr7\n",
      "\tExtracting 651 sequences from chr8\n",
      "\tExtracting 860 sequences from chr9\n",
      "\tExtracting 667 sequences from chr10\n",
      "\tExtracting 1356 sequences from chr11\n",
      "\tExtracting 1188 sequences from chr12\n",
      "\tExtracting 253 sequences from chr13\n",
      "\tExtracting 697 sequences from chr14\n",
      "\tExtracting 555 sequences from chr15\n",
      "\tExtracting 1456 sequences from chr16\n",
      "\tExtracting 1970 sequences from chr17\n",
      "\tExtracting 218 sequences from chr18\n",
      "\tExtracting 1831 sequences from chr19\n",
      "\tExtracting 636 sequences from chr20\n",
      "\tExtracting 308 sequences from chr21\n",
      "\tExtracting 879 sequences from chr22\n",
      "\tExtracting 701 sequences from chrX\n",
      "\tExtracting 41 sequences from chrM\n",
      "\t265 of 22370 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t22105 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 34934\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 34934\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 38 (avg size of targets)\n",
      "\tBackground files for 38 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 3680 sequences from chr1\n",
      "\tExtracting 2093 sequences from chr2\n",
      "\tExtracting 1870 sequences from chr3\n",
      "\tExtracting 886 sequences from chr4\n",
      "\tExtracting 1304 sequences from chr5\n",
      "\tExtracting 1539 sequences from chr6\n",
      "\tExtracting 1591 sequences from chr7\n",
      "\tExtracting 1063 sequences from chr8\n",
      "\tExtracting 1301 sequences from chr9\n",
      "\tExtracting 1063 sequences from chr10\n",
      "\tExtracting 2045 sequences from chr11\n",
      "\tExtracting 1990 sequences from chr12\n",
      "\tExtracting 460 sequences from chr13\n",
      "\tExtracting 1147 sequences from chr14\n",
      "\tExtracting 1047 sequences from chr15\n",
      "\tExtracting 2210 sequences from chr16\n",
      "\tExtracting 2958 sequences from chr17\n",
      "\tExtracting 419 sequences from chr18\n",
      "\tExtracting 2522 sequences from chr19\n",
      "\tExtracting 875 sequences from chr20\n",
      "\tExtracting 427 sequences from chr21\n",
      "\tExtracting 1252 sequences from chr22\n",
      "\tExtracting 1145 sequences from chrX\n",
      "\tExtracting 47 sequences from chrM\n",
      "\t419 of 34934 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t34515 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 27038\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 27038\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 31 (avg size of targets)\n",
      "\tBackground files for 31 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2806 sequences from chr1\n",
      "\tExtracting 1609 sequences from chr2\n",
      "\tExtracting 1452 sequences from chr3\n",
      "\tExtracting 751 sequences from chr4\n",
      "\tExtracting 978 sequences from chr5\n",
      "\tExtracting 1211 sequences from chr6\n",
      "\tExtracting 1291 sequences from chr7\n",
      "\tExtracting 760 sequences from chr8\n",
      "\tExtracting 1130 sequences from chr9\n",
      "\tExtracting 925 sequences from chr10\n",
      "\tExtracting 1442 sequences from chr11\n",
      "\tExtracting 1359 sequences from chr12\n",
      "\tExtracting 424 sequences from chr13\n",
      "\tExtracting 760 sequences from chr14\n",
      "\tExtracting 786 sequences from chr15\n",
      "\tExtracting 1571 sequences from chr16\n",
      "\tExtracting 2054 sequences from chr17\n",
      "\tExtracting 347 sequences from chr18\n",
      "\tExtracting 2101 sequences from chr19\n",
      "\tExtracting 816 sequences from chr20\n",
      "\tExtracting 423 sequences from chr21\n",
      "\tExtracting 1114 sequences from chr22\n",
      "\tExtracting 877 sequences from chrX\n",
      "\tExtracting 51 sequences from chrM\n",
      "\t255 of 27038 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t26783 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 33665\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 33665\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 41 (avg size of targets)\n",
      "\tBackground files for 41 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 3475 sequences from chr1\n",
      "\tExtracting 1969 sequences from chr2\n",
      "\tExtracting 1793 sequences from chr3\n",
      "\tExtracting 899 sequences from chr4\n",
      "\tExtracting 1280 sequences from chr5\n",
      "\tExtracting 1362 sequences from chr6\n",
      "\tExtracting 1600 sequences from chr7\n",
      "\tExtracting 1056 sequences from chr8\n",
      "\tExtracting 1349 sequences from chr9\n",
      "\tExtracting 1069 sequences from chr10\n",
      "\tExtracting 1956 sequences from chr11\n",
      "\tExtracting 1706 sequences from chr12\n",
      "\tExtracting 437 sequences from chr13\n",
      "\tExtracting 1020 sequences from chr14\n",
      "\tExtracting 966 sequences from chr15\n",
      "\tExtracting 2149 sequences from chr16\n",
      "\tExtracting 2802 sequences from chr17\n",
      "\tExtracting 415 sequences from chr18\n",
      "\tExtracting 2622 sequences from chr19\n",
      "\tExtracting 922 sequences from chr20\n",
      "\tExtracting 466 sequences from chr21\n",
      "\tExtracting 1309 sequences from chr22\n",
      "\tExtracting 1003 sequences from chrX\n",
      "\tExtracting 40 sequences from chrM\n",
      "\t277 of 33665 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t33388 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 28458\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 28458\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 41 (avg size of targets)\n",
      "\tBackground files for 41 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2986 sequences from chr1\n",
      "\tExtracting 1655 sequences from chr2\n",
      "\tExtracting 1472 sequences from chr3\n",
      "\tExtracting 740 sequences from chr4\n",
      "\tExtracting 1065 sequences from chr5\n",
      "\tExtracting 1208 sequences from chr6\n",
      "\tExtracting 1317 sequences from chr7\n",
      "\tExtracting 848 sequences from chr8\n",
      "\tExtracting 1088 sequences from chr9\n",
      "\tExtracting 863 sequences from chr10\n",
      "\tExtracting 1658 sequences from chr11\n",
      "\tExtracting 1496 sequences from chr12\n",
      "\tExtracting 348 sequences from chr13\n",
      "\tExtracting 867 sequences from chr14\n",
      "\tExtracting 750 sequences from chr15\n",
      "\tExtracting 1906 sequences from chr16\n",
      "\tExtracting 2406 sequences from chr17\n",
      "\tExtracting 341 sequences from chr18\n",
      "\tExtracting 2245 sequences from chr19\n",
      "\tExtracting 789 sequences from chr20\n",
      "\tExtracting 374 sequences from chr21\n",
      "\tExtracting 1103 sequences from chr22\n",
      "\tExtracting 887 sequences from chrX\n",
      "\tExtracting 46 sequences from chrM\n",
      "\t270 of 28458 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t28188 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 20718\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 20718\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 33 (avg size of targets)\n",
      "\tBackground files for 33 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2019 sequences from chr1\n",
      "\tExtracting 1258 sequences from chr2\n",
      "\tExtracting 1165 sequences from chr3\n",
      "\tExtracting 621 sequences from chr4\n",
      "\tExtracting 721 sequences from chr5\n",
      "\tExtracting 958 sequences from chr6\n",
      "\tExtracting 988 sequences from chr7\n",
      "\tExtracting 604 sequences from chr8\n",
      "\tExtracting 882 sequences from chr9\n",
      "\tExtracting 760 sequences from chr10\n",
      "\tExtracting 1026 sequences from chr11\n",
      "\tExtracting 876 sequences from chr12\n",
      "\tExtracting 400 sequences from chr13\n",
      "\tExtracting 575 sequences from chr14\n",
      "\tExtracting 556 sequences from chr15\n",
      "\tExtracting 1041 sequences from chr16\n",
      "\tExtracting 1568 sequences from chr17\n",
      "\tExtracting 274 sequences from chr18\n",
      "\tExtracting 1710 sequences from chr19\n",
      "\tExtracting 654 sequences from chr20\n",
      "\tExtracting 341 sequences from chr21\n",
      "\tExtracting 946 sequences from chr22\n",
      "\tExtracting 733 sequences from chrX\n",
      "\tExtracting 42 sequences from chrM\n",
      "\t118 of 20718 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t20600 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /home/gpratt/projects/encode/analysis/encode_v8/assigned/RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed.all.real.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 27088\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 27088\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 38 (avg size of targets)\n",
      "\tBackground files for 38 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2873 sequences from chr1\n",
      "\tExtracting 1574 sequences from chr2\n",
      "\tExtracting 1370 sequences from chr3\n",
      "\tExtracting 708 sequences from chr4\n",
      "\tExtracting 1001 sequences from chr5\n",
      "\tExtracting 1125 sequences from chr6\n",
      "\tExtracting 1287 sequences from chr7\n",
      "\tExtracting 783 sequences from chr8\n",
      "\tExtracting 1094 sequences from chr9\n",
      "\tExtracting 849 sequences from chr10\n",
      "\tExtracting 1541 sequences from chr11\n",
      "\tExtracting 1420 sequences from chr12\n",
      "\tExtracting 326 sequences from chr13\n",
      "\tExtracting 826 sequences from chr14\n",
      "\tExtracting 713 sequences from chr15\n",
      "\tExtracting 1772 sequences from chr16\n",
      "\tExtracting 2280 sequences from chr17\n",
      "\tExtracting 271 sequences from chr18\n",
      "\tExtracting 2177 sequences from chr19\n",
      "\tExtracting 780 sequences from chr20\n",
      "\tExtracting 381 sequences from chr21\n",
      "\tExtracting 1081 sequences from chr22\n",
      "\tExtracting 819 sequences from chrX\n",
      "\tExtracting 37 sequences from chrM\n",
      "\t239 of 27088 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t26849 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 33194\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 33194\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 38 (avg size of targets)\n",
      "\tBackground files for 38 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 3131 sequences from chr1\n",
      "\tExtracting 2592 sequences from chr2\n",
      "\tExtracting 1993 sequences from chr3\n",
      "\tExtracting 1424 sequences from chr4\n",
      "\tExtracting 1549 sequences from chr5\n",
      "\tExtracting 1535 sequences from chr6\n",
      "\tExtracting 1653 sequences from chr7\n",
      "\tExtracting 1252 sequences from chr8\n",
      "\tExtracting 1281 sequences from chr9\n",
      "\tExtracting 1324 sequences from chr10\n",
      "\tExtracting 1746 sequences from chr11\n",
      "\tExtracting 1885 sequences from chr12\n",
      "\tExtracting 620 sequences from chr13\n",
      "\tExtracting 1170 sequences from chr14\n",
      "\tExtracting 1268 sequences from chr15\n",
      "\tExtracting 1423 sequences from chr16\n",
      "\tExtracting 1940 sequences from chr17\n",
      "\tExtracting 593 sequences from chr18\n",
      "\tExtracting 1773 sequences from chr19\n",
      "\tExtracting 775 sequences from chr20\n",
      "\tExtracting 328 sequences from chr21\n",
      "\tExtracting 708 sequences from chr22\n",
      "\tExtracting 1071 sequences from chrX\n",
      "\tExtracting 159 sequences from chrY\n",
      "\tExtracting 1 sequences from chrM\n",
      "\t511 of 33194 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t32683 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 914\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 914\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 35 (avg size of targets)\n",
      "\tBackground files for 35 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 83 sequences from chr1\n",
      "\tExtracting 71 sequences from chr2\n",
      "\tExtracting 58 sequences from chr3\n",
      "\tExtracting 42 sequences from chr4\n",
      "\tExtracting 36 sequences from chr5\n",
      "\tExtracting 43 sequences from chr6\n",
      "\tExtracting 43 sequences from chr7\n",
      "\tExtracting 33 sequences from chr8\n",
      "\tExtracting 36 sequences from chr9\n",
      "\tExtracting 47 sequences from chr10\n",
      "\tExtracting 48 sequences from chr11\n",
      "\tExtracting 45 sequences from chr12\n",
      "\tExtracting 20 sequences from chr13\n",
      "\tExtracting 32 sequences from chr14\n",
      "\tExtracting 37 sequences from chr15\n",
      "\tExtracting 39 sequences from chr16\n",
      "\tExtracting 53 sequences from chr17\n",
      "\tExtracting 19 sequences from chr18\n",
      "\tExtracting 39 sequences from chr19\n",
      "\tExtracting 23 sequences from chr20\n",
      "\tExtracting 14 sequences from chr21\n",
      "\tExtracting 20 sequences from chr22\n",
      "\tExtracting 27 sequences from chrX\n",
      "\tExtracting 6 sequences from chrY\n",
      "\t1 of 914 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t913 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 20495\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 20495\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 33 (avg size of targets)\n",
      "\tBackground files for 33 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 1919 sequences from chr1\n",
      "\tExtracting 1608 sequences from chr2\n",
      "\tExtracting 1220 sequences from chr3\n",
      "\tExtracting 895 sequences from chr4\n",
      "\tExtracting 987 sequences from chr5\n",
      "\tExtracting 983 sequences from chr6\n",
      "\tExtracting 998 sequences from chr7\n",
      "\tExtracting 783 sequences from chr8\n",
      "\tExtracting 801 sequences from chr9\n",
      "\tExtracting 820 sequences from chr10\n",
      "\tExtracting 1090 sequences from chr11\n",
      "\tExtracting 1120 sequences from chr12\n",
      "\tExtracting 401 sequences from chr13\n",
      "\tExtracting 679 sequences from chr14\n",
      "\tExtracting 731 sequences from chr15\n",
      "\tExtracting 872 sequences from chr16\n",
      "\tExtracting 1142 sequences from chr17\n",
      "\tExtracting 380 sequences from chr18\n",
      "\tExtracting 1146 sequences from chr19\n",
      "\tExtracting 474 sequences from chr20\n",
      "\tExtracting 221 sequences from chr21\n",
      "\tExtracting 482 sequences from chr22\n",
      "\tExtracting 649 sequences from chrX\n",
      "\tExtracting 94 sequences from chrY\n",
      "\t277 of 20495 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t20218 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 22370\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 22370\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 36 (avg size of targets)\n",
      "\tBackground files for 36 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2174 sequences from chr1\n",
      "\tExtracting 1697 sequences from chr2\n",
      "\tExtracting 1362 sequences from chr3\n",
      "\tExtracting 985 sequences from chr4\n",
      "\tExtracting 1012 sequences from chr5\n",
      "\tExtracting 1095 sequences from chr6\n",
      "\tExtracting 1126 sequences from chr7\n",
      "\tExtracting 878 sequences from chr8\n",
      "\tExtracting 853 sequences from chr9\n",
      "\tExtracting 884 sequences from chr10\n",
      "\tExtracting 1170 sequences from chr11\n",
      "\tExtracting 1252 sequences from chr12\n",
      "\tExtracting 408 sequences from chr13\n",
      "\tExtracting 732 sequences from chr14\n",
      "\tExtracting 854 sequences from chr15\n",
      "\tExtracting 951 sequences from chr16\n",
      "\tExtracting 1288 sequences from chr17\n",
      "\tExtracting 375 sequences from chr18\n",
      "\tExtracting 1204 sequences from chr19\n",
      "\tExtracting 509 sequences from chr20\n",
      "\tExtracting 233 sequences from chr21\n",
      "\tExtracting 481 sequences from chr22\n",
      "\tExtracting 752 sequences from chrX\n",
      "\tExtracting 95 sequences from chrY\n",
      "\t265 of 22370 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t22105 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 34934\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 34934\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 38 (avg size of targets)\n",
      "\tBackground files for 38 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 3252 sequences from chr1\n",
      "\tExtracting 2742 sequences from chr2\n",
      "\tExtracting 2082 sequences from chr3\n",
      "\tExtracting 1484 sequences from chr4\n",
      "\tExtracting 1643 sequences from chr5\n",
      "\tExtracting 1673 sequences from chr6\n",
      "\tExtracting 1774 sequences from chr7\n",
      "\tExtracting 1350 sequences from chr8\n",
      "\tExtracting 1380 sequences from chr9\n",
      "\tExtracting 1394 sequences from chr10\n",
      "\tExtracting 1907 sequences from chr11\n",
      "\tExtracting 1985 sequences from chr12\n",
      "\tExtracting 683 sequences from chr13\n",
      "\tExtracting 1131 sequences from chr14\n",
      "\tExtracting 1325 sequences from chr15\n",
      "\tExtracting 1552 sequences from chr16\n",
      "\tExtracting 1972 sequences from chr17\n",
      "\tExtracting 592 sequences from chr18\n",
      "\tExtracting 1789 sequences from chr19\n",
      "\tExtracting 745 sequences from chr20\n",
      "\tExtracting 395 sequences from chr21\n",
      "\tExtracting 753 sequences from chr22\n",
      "\tExtracting 1137 sequences from chrX\n",
      "\tExtracting 194 sequences from chrY\n",
      "\t419 of 34934 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t34515 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 27038\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 27038\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 31 (avg size of targets)\n",
      "\tBackground files for 31 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2534 sequences from chr1\n",
      "\tExtracting 2118 sequences from chr2\n",
      "\tExtracting 1651 sequences from chr3\n",
      "\tExtracting 1205 sequences from chr4\n",
      "\tExtracting 1350 sequences from chr5\n",
      "\tExtracting 1310 sequences from chr6\n",
      "\tExtracting 1325 sequences from chr7\n",
      "\tExtracting 1069 sequences from chr8\n",
      "\tExtracting 1074 sequences from chr9\n",
      "\tExtracting 1109 sequences from chr10\n",
      "\tExtracting 1345 sequences from chr11\n",
      "\tExtracting 1487 sequences from chr12\n",
      "\tExtracting 548 sequences from chr13\n",
      "\tExtracting 889 sequences from chr14\n",
      "\tExtracting 1051 sequences from chr15\n",
      "\tExtracting 1169 sequences from chr16\n",
      "\tExtracting 1475 sequences from chr17\n",
      "\tExtracting 491 sequences from chr18\n",
      "\tExtracting 1344 sequences from chr19\n",
      "\tExtracting 632 sequences from chr20\n",
      "\tExtracting 325 sequences from chr21\n",
      "\tExtracting 579 sequences from chr22\n",
      "\tExtracting 811 sequences from chrX\n",
      "\tExtracting 147 sequences from chrY\n",
      "\t255 of 27038 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t26783 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 33665\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 33665\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 41 (avg size of targets)\n",
      "\tBackground files for 41 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 3216 sequences from chr1\n",
      "\tExtracting 2584 sequences from chr2\n",
      "\tExtracting 2070 sequences from chr3\n",
      "\tExtracting 1486 sequences from chr4\n",
      "\tExtracting 1551 sequences from chr5\n",
      "\tExtracting 1588 sequences from chr6\n",
      "\tExtracting 1743 sequences from chr7\n",
      "\tExtracting 1280 sequences from chr8\n",
      "\tExtracting 1312 sequences from chr9\n",
      "\tExtracting 1424 sequences from chr10\n",
      "\tExtracting 1719 sequences from chr11\n",
      "\tExtracting 1856 sequences from chr12\n",
      "\tExtracting 640 sequences from chr13\n",
      "\tExtracting 1134 sequences from chr14\n",
      "\tExtracting 1305 sequences from chr15\n",
      "\tExtracting 1442 sequences from chr16\n",
      "\tExtracting 1829 sequences from chr17\n",
      "\tExtracting 606 sequences from chr18\n",
      "\tExtracting 1755 sequences from chr19\n",
      "\tExtracting 783 sequences from chr20\n",
      "\tExtracting 351 sequences from chr21\n",
      "\tExtracting 716 sequences from chr22\n",
      "\tExtracting 1096 sequences from chrX\n",
      "\tExtracting 175 sequences from chrY\n",
      "\tExtracting 4 sequences from chrM\n",
      "\t277 of 33665 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t33388 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 28458\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 28458\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 41 (avg size of targets)\n",
      "\tBackground files for 41 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2616 sequences from chr1\n",
      "\tExtracting 2247 sequences from chr2\n",
      "\tExtracting 1671 sequences from chr3\n",
      "\tExtracting 1213 sequences from chr4\n",
      "\tExtracting 1381 sequences from chr5\n",
      "\tExtracting 1400 sequences from chr6\n",
      "\tExtracting 1434 sequences from chr7\n",
      "\tExtracting 1118 sequences from chr8\n",
      "\tExtracting 1044 sequences from chr9\n",
      "\tExtracting 1199 sequences from chr10\n",
      "\tExtracting 1434 sequences from chr11\n",
      "\tExtracting 1634 sequences from chr12\n",
      "\tExtracting 573 sequences from chr13\n",
      "\tExtracting 912 sequences from chr14\n",
      "\tExtracting 1046 sequences from chr15\n",
      "\tExtracting 1262 sequences from chr16\n",
      "\tExtracting 1581 sequences from chr17\n",
      "\tExtracting 552 sequences from chr18\n",
      "\tExtracting 1491 sequences from chr19\n",
      "\tExtracting 693 sequences from chr20\n",
      "\tExtracting 329 sequences from chr21\n",
      "\tExtracting 600 sequences from chr22\n",
      "\tExtracting 892 sequences from chrX\n",
      "\tExtracting 136 sequences from chrY\n",
      "\t270 of 28458 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t28188 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 20718\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 20718\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 33 (avg size of targets)\n",
      "\tBackground files for 33 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 1975 sequences from chr1\n",
      "\tExtracting 1559 sequences from chr2\n",
      "\tExtracting 1267 sequences from chr3\n",
      "\tExtracting 947 sequences from chr4\n",
      "\tExtracting 1020 sequences from chr5\n",
      "\tExtracting 1004 sequences from chr6\n",
      "\tExtracting 1092 sequences from chr7\n",
      "\tExtracting 803 sequences from chr8\n",
      "\tExtracting 849 sequences from chr9\n",
      "\tExtracting 863 sequences from chr10\n",
      "\tExtracting 1006 sequences from chr11\n",
      "\tExtracting 1157 sequences from chr12\n",
      "\tExtracting 443 sequences from chr13\n",
      "\tExtracting 667 sequences from chr14\n",
      "\tExtracting 795 sequences from chr15\n",
      "\tExtracting 867 sequences from chr16\n",
      "\tExtracting 1054 sequences from chr17\n",
      "\tExtracting 410 sequences from chr18\n",
      "\tExtracting 991 sequences from chr19\n",
      "\tExtracting 500 sequences from chr20\n",
      "\tExtracting 251 sequences from chr21\n",
      "\tExtracting 444 sequences from chr22\n",
      "\tExtracting 610 sequences from chrX\n",
      "\tExtracting 144 sequences from chrY\n",
      "\t118 of 20718 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t20600 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed.all.rand.1.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 27088\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 27088\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 38 (avg size of targets)\n",
      "\tBackground files for 38 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2600 sequences from chr1\n",
      "\tExtracting 2015 sequences from chr2\n",
      "\tExtracting 1632 sequences from chr3\n",
      "\tExtracting 1174 sequences from chr4\n",
      "\tExtracting 1280 sequences from chr5\n",
      "\tExtracting 1352 sequences from chr6\n",
      "\tExtracting 1352 sequences from chr7\n",
      "\tExtracting 1038 sequences from chr8\n",
      "\tExtracting 973 sequences from chr9\n",
      "\tExtracting 1153 sequences from chr10\n",
      "\tExtracting 1412 sequences from chr11\n",
      "\tExtracting 1575 sequences from chr12\n",
      "\tExtracting 521 sequences from chr13\n",
      "\tExtracting 931 sequences from chr14\n",
      "\tExtracting 1015 sequences from chr15\n",
      "\tExtracting 1130 sequences from chr16\n",
      "\tExtracting 1473 sequences from chr17\n",
      "\tExtracting 500 sequences from chr18\n",
      "\tExtracting 1483 sequences from chr19\n",
      "\tExtracting 608 sequences from chr20\n",
      "\tExtracting 298 sequences from chr21\n",
      "\tExtracting 544 sequences from chr22\n",
      "\tExtracting 901 sequences from chrX\n",
      "\tExtracting 128 sequences from chrY\n",
      "\t239 of 27088 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t26849 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 33194\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 33194\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 38 (avg size of targets)\n",
      "\tBackground files for 38 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 3111 sequences from chr1\n",
      "\tExtracting 2551 sequences from chr2\n",
      "\tExtracting 1971 sequences from chr3\n",
      "\tExtracting 1444 sequences from chr4\n",
      "\tExtracting 1500 sequences from chr5\n",
      "\tExtracting 1528 sequences from chr6\n",
      "\tExtracting 1630 sequences from chr7\n",
      "\tExtracting 1232 sequences from chr8\n",
      "\tExtracting 1319 sequences from chr9\n",
      "\tExtracting 1385 sequences from chr10\n",
      "\tExtracting 1805 sequences from chr11\n",
      "\tExtracting 1922 sequences from chr12\n",
      "\tExtracting 557 sequences from chr13\n",
      "\tExtracting 1061 sequences from chr14\n",
      "\tExtracting 1291 sequences from chr15\n",
      "\tExtracting 1479 sequences from chr16\n",
      "\tExtracting 1905 sequences from chr17\n",
      "\tExtracting 607 sequences from chr18\n",
      "\tExtracting 1848 sequences from chr19\n",
      "\tExtracting 760 sequences from chr20\n",
      "\tExtracting 390 sequences from chr21\n",
      "\tExtracting 696 sequences from chr22\n",
      "\tExtracting 1030 sequences from chrX\n",
      "\tExtracting 171 sequences from chrY\n",
      "\tExtracting 1 sequences from chrM\n",
      "\t511 of 33194 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t32683 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 914\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 914\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 35 (avg size of targets)\n",
      "\tBackground files for 35 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 83 sequences from chr1\n",
      "\tExtracting 74 sequences from chr2\n",
      "\tExtracting 55 sequences from chr3\n",
      "\tExtracting 47 sequences from chr4\n",
      "\tExtracting 40 sequences from chr5\n",
      "\tExtracting 48 sequences from chr6\n",
      "\tExtracting 35 sequences from chr7\n",
      "\tExtracting 46 sequences from chr8\n",
      "\tExtracting 33 sequences from chr9\n",
      "\tExtracting 38 sequences from chr10\n",
      "\tExtracting 35 sequences from chr11\n",
      "\tExtracting 45 sequences from chr12\n",
      "\tExtracting 15 sequences from chr13\n",
      "\tExtracting 30 sequences from chr14\n",
      "\tExtracting 39 sequences from chr15\n",
      "\tExtracting 36 sequences from chr16\n",
      "\tExtracting 58 sequences from chr17\n",
      "\tExtracting 21 sequences from chr18\n",
      "\tExtracting 37 sequences from chr19\n",
      "\tExtracting 24 sequences from chr20\n",
      "\tExtracting 12 sequences from chr21\n",
      "\tExtracting 27 sequences from chr22\n",
      "\tExtracting 30 sequences from chrX\n",
      "\tExtracting 6 sequences from chrY\n",
      "\t1 of 914 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t913 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 20495\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 20495\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 33 (avg size of targets)\n",
      "\tBackground files for 33 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 1892 sequences from chr1\n",
      "\tExtracting 1608 sequences from chr2\n",
      "\tExtracting 1281 sequences from chr3\n",
      "\tExtracting 869 sequences from chr4\n",
      "\tExtracting 953 sequences from chr5\n",
      "\tExtracting 982 sequences from chr6\n",
      "\tExtracting 960 sequences from chr7\n",
      "\tExtracting 749 sequences from chr8\n",
      "\tExtracting 780 sequences from chr9\n",
      "\tExtracting 830 sequences from chr10\n",
      "\tExtracting 1130 sequences from chr11\n",
      "\tExtracting 1189 sequences from chr12\n",
      "\tExtracting 378 sequences from chr13\n",
      "\tExtracting 708 sequences from chr14\n",
      "\tExtracting 797 sequences from chr15\n",
      "\tExtracting 896 sequences from chr16\n",
      "\tExtracting 1148 sequences from chr17\n",
      "\tExtracting 385 sequences from chr18\n",
      "\tExtracting 1103 sequences from chr19\n",
      "\tExtracting 494 sequences from chr20\n",
      "\tExtracting 225 sequences from chr21\n",
      "\tExtracting 405 sequences from chr22\n",
      "\tExtracting 635 sequences from chrX\n",
      "\tExtracting 97 sequences from chrY\n",
      "\tExtracting 1 sequences from chrM\n",
      "\t277 of 20495 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t20218 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 22370\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 22370\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 36 (avg size of targets)\n",
      "\tBackground files for 36 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2130 sequences from chr1\n",
      "\tExtracting 1736 sequences from chr2\n",
      "\tExtracting 1359 sequences from chr3\n",
      "\tExtracting 948 sequences from chr4\n",
      "\tExtracting 1089 sequences from chr5\n",
      "\tExtracting 1036 sequences from chr6\n",
      "\tExtracting 1147 sequences from chr7\n",
      "\tExtracting 845 sequences from chr8\n",
      "\tExtracting 886 sequences from chr9\n",
      "\tExtracting 886 sequences from chr10\n",
      "\tExtracting 1207 sequences from chr11\n",
      "\tExtracting 1284 sequences from chr12\n",
      "\tExtracting 399 sequences from chr13\n",
      "\tExtracting 789 sequences from chr14\n",
      "\tExtracting 833 sequences from chr15\n",
      "\tExtracting 1010 sequences from chr16\n",
      "\tExtracting 1144 sequences from chr17\n",
      "\tExtracting 416 sequences from chr18\n",
      "\tExtracting 1194 sequences from chr19\n",
      "\tExtracting 460 sequences from chr20\n",
      "\tExtracting 252 sequences from chr21\n",
      "\tExtracting 494 sequences from chr22\n",
      "\tExtracting 717 sequences from chrX\n",
      "\tExtracting 108 sequences from chrY\n",
      "\tExtracting 1 sequences from chrM\n",
      "\t265 of 22370 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t22105 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 34934\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 34934\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 38 (avg size of targets)\n",
      "\tBackground files for 38 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 3228 sequences from chr1\n",
      "\tExtracting 2691 sequences from chr2\n",
      "\tExtracting 2141 sequences from chr3\n",
      "\tExtracting 1544 sequences from chr4\n",
      "\tExtracting 1616 sequences from chr5\n",
      "\tExtracting 1671 sequences from chr6\n",
      "\tExtracting 1770 sequences from chr7\n",
      "\tExtracting 1298 sequences from chr8\n",
      "\tExtracting 1330 sequences from chr9\n",
      "\tExtracting 1418 sequences from chr10\n",
      "\tExtracting 1826 sequences from chr11\n",
      "\tExtracting 2032 sequences from chr12\n",
      "\tExtracting 650 sequences from chr13\n",
      "\tExtracting 1173 sequences from chr14\n",
      "\tExtracting 1352 sequences from chr15\n",
      "\tExtracting 1521 sequences from chr16\n",
      "\tExtracting 1914 sequences from chr17\n",
      "\tExtracting 657 sequences from chr18\n",
      "\tExtracting 1888 sequences from chr19\n",
      "\tExtracting 817 sequences from chr20\n",
      "\tExtracting 367 sequences from chr21\n",
      "\tExtracting 761 sequences from chr22\n",
      "\tExtracting 1083 sequences from chrX\n",
      "\tExtracting 185 sequences from chrY\n",
      "\tExtracting 1 sequences from chrM\n",
      "\t419 of 34934 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t34515 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 27038\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 27038\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 31 (avg size of targets)\n",
      "\tBackground files for 31 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2427 sequences from chr1\n",
      "\tExtracting 2128 sequences from chr2\n",
      "\tExtracting 1666 sequences from chr3\n",
      "\tExtracting 1202 sequences from chr4\n",
      "\tExtracting 1263 sequences from chr5\n",
      "\tExtracting 1301 sequences from chr6\n",
      "\tExtracting 1343 sequences from chr7\n",
      "\tExtracting 1053 sequences from chr8\n",
      "\tExtracting 1119 sequences from chr9\n",
      "\tExtracting 1149 sequences from chr10\n",
      "\tExtracting 1388 sequences from chr11\n",
      "\tExtracting 1579 sequences from chr12\n",
      "\tExtracting 555 sequences from chr13\n",
      "\tExtracting 868 sequences from chr14\n",
      "\tExtracting 1006 sequences from chr15\n",
      "\tExtracting 1230 sequences from chr16\n",
      "\tExtracting 1357 sequences from chr17\n",
      "\tExtracting 509 sequences from chr18\n",
      "\tExtracting 1378 sequences from chr19\n",
      "\tExtracting 622 sequences from chr20\n",
      "\tExtracting 322 sequences from chr21\n",
      "\tExtracting 587 sequences from chr22\n",
      "\tExtracting 833 sequences from chrX\n",
      "\tExtracting 153 sequences from chrY\n",
      "\t255 of 27038 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t26783 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 33665\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 33665\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 41 (avg size of targets)\n",
      "\tBackground files for 41 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 3218 sequences from chr1\n",
      "\tExtracting 2548 sequences from chr2\n",
      "\tExtracting 2023 sequences from chr3\n",
      "\tExtracting 1529 sequences from chr4\n",
      "\tExtracting 1681 sequences from chr5\n",
      "\tExtracting 1551 sequences from chr6\n",
      "\tExtracting 1689 sequences from chr7\n",
      "\tExtracting 1253 sequences from chr8\n",
      "\tExtracting 1307 sequences from chr9\n",
      "\tExtracting 1460 sequences from chr10\n",
      "\tExtracting 1696 sequences from chr11\n",
      "\tExtracting 1901 sequences from chr12\n",
      "\tExtracting 672 sequences from chr13\n",
      "\tExtracting 1074 sequences from chr14\n",
      "\tExtracting 1289 sequences from chr15\n",
      "\tExtracting 1490 sequences from chr16\n",
      "\tExtracting 1844 sequences from chr17\n",
      "\tExtracting 628 sequences from chr18\n",
      "\tExtracting 1785 sequences from chr19\n",
      "\tExtracting 724 sequences from chr20\n",
      "\tExtracting 367 sequences from chr21\n",
      "\tExtracting 708 sequences from chr22\n",
      "\tExtracting 1040 sequences from chrX\n",
      "\tExtracting 188 sequences from chrY\n",
      "\t277 of 33665 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t33388 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 28458\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 28458\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 41 (avg size of targets)\n",
      "\tBackground files for 41 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2693 sequences from chr1\n",
      "\tExtracting 2206 sequences from chr2\n",
      "\tExtracting 1703 sequences from chr3\n",
      "\tExtracting 1314 sequences from chr4\n",
      "\tExtracting 1374 sequences from chr5\n",
      "\tExtracting 1316 sequences from chr6\n",
      "\tExtracting 1441 sequences from chr7\n",
      "\tExtracting 1030 sequences from chr8\n",
      "\tExtracting 1089 sequences from chr9\n",
      "\tExtracting 1135 sequences from chr10\n",
      "\tExtracting 1403 sequences from chr11\n",
      "\tExtracting 1587 sequences from chr12\n",
      "\tExtracting 521 sequences from chr13\n",
      "\tExtracting 966 sequences from chr14\n",
      "\tExtracting 1061 sequences from chr15\n",
      "\tExtracting 1256 sequences from chr16\n",
      "\tExtracting 1632 sequences from chr17\n",
      "\tExtracting 506 sequences from chr18\n",
      "\tExtracting 1535 sequences from chr19\n",
      "\tExtracting 671 sequences from chr20\n",
      "\tExtracting 346 sequences from chr21\n",
      "\tExtracting 625 sequences from chr22\n",
      "\tExtracting 911 sequences from chrX\n",
      "\tExtracting 137 sequences from chrY\n",
      "\t270 of 28458 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t28188 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 20718\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 20718\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 33 (avg size of targets)\n",
      "\tBackground files for 33 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 1961 sequences from chr1\n",
      "\tExtracting 1639 sequences from chr2\n",
      "\tExtracting 1228 sequences from chr3\n",
      "\tExtracting 971 sequences from chr4\n",
      "\tExtracting 993 sequences from chr5\n",
      "\tExtracting 951 sequences from chr6\n",
      "\tExtracting 1082 sequences from chr7\n",
      "\tExtracting 796 sequences from chr8\n",
      "\tExtracting 790 sequences from chr9\n",
      "\tExtracting 884 sequences from chr10\n",
      "\tExtracting 1008 sequences from chr11\n",
      "\tExtracting 1115 sequences from chr12\n",
      "\tExtracting 472 sequences from chr13\n",
      "\tExtracting 702 sequences from chr14\n",
      "\tExtracting 789 sequences from chr15\n",
      "\tExtracting 875 sequences from chr16\n",
      "\tExtracting 1047 sequences from chr17\n",
      "\tExtracting 396 sequences from chr18\n",
      "\tExtracting 1045 sequences from chr19\n",
      "\tExtracting 536 sequences from chr20\n",
      "\tExtracting 243 sequences from chr21\n",
      "\tExtracting 418 sequences from chr22\n",
      "\tExtracting 638 sequences from chrX\n",
      "\tExtracting 139 sequences from chrY\n",
      "\t118 of 20718 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t20600 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed.all.rand.2.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 27088\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 27088\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 38 (avg size of targets)\n",
      "\tBackground files for 38 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2626 sequences from chr1\n",
      "\tExtracting 2058 sequences from chr2\n",
      "\tExtracting 1623 sequences from chr3\n",
      "\tExtracting 1199 sequences from chr4\n",
      "\tExtracting 1268 sequences from chr5\n",
      "\tExtracting 1328 sequences from chr6\n",
      "\tExtracting 1362 sequences from chr7\n",
      "\tExtracting 1004 sequences from chr8\n",
      "\tExtracting 1081 sequences from chr9\n",
      "\tExtracting 1084 sequences from chr10\n",
      "\tExtracting 1309 sequences from chr11\n",
      "\tExtracting 1618 sequences from chr12\n",
      "\tExtracting 508 sequences from chr13\n",
      "\tExtracting 905 sequences from chr14\n",
      "\tExtracting 1008 sequences from chr15\n",
      "\tExtracting 1108 sequences from chr16\n",
      "\tExtracting 1539 sequences from chr17\n",
      "\tExtracting 513 sequences from chr18\n",
      "\tExtracting 1424 sequences from chr19\n",
      "\tExtracting 634 sequences from chr20\n",
      "\tExtracting 317 sequences from chr21\n",
      "\tExtracting 569 sequences from chr22\n",
      "\tExtracting 861 sequences from chrX\n",
      "\tExtracting 142 sequences from chrY\n",
      "\t239 of 27088 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t26849 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP10_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 33194\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 33194\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 38 (avg size of targets)\n",
      "\tBackground files for 38 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 3089 sequences from chr1\n",
      "\tExtracting 2584 sequences from chr2\n",
      "\tExtracting 2038 sequences from chr3\n",
      "\tExtracting 1393 sequences from chr4\n",
      "\tExtracting 1610 sequences from chr5\n",
      "\tExtracting 1578 sequences from chr6\n",
      "\tExtracting 1642 sequences from chr7\n",
      "\tExtracting 1241 sequences from chr8\n",
      "\tExtracting 1259 sequences from chr9\n",
      "\tExtracting 1343 sequences from chr10\n",
      "\tExtracting 1673 sequences from chr11\n",
      "\tExtracting 2005 sequences from chr12\n",
      "\tExtracting 622 sequences from chr13\n",
      "\tExtracting 1032 sequences from chr14\n",
      "\tExtracting 1249 sequences from chr15\n",
      "\tExtracting 1463 sequences from chr16\n",
      "\tExtracting 1879 sequences from chr17\n",
      "\tExtracting 611 sequences from chr18\n",
      "\tExtracting 1771 sequences from chr19\n",
      "\tExtracting 790 sequences from chr20\n",
      "\tExtracting 390 sequences from chr21\n",
      "\tExtracting 726 sequences from chr22\n",
      "\tExtracting 1059 sequences from chrX\n",
      "\tExtracting 146 sequences from chrY\n",
      "\tExtracting 1 sequences from chrM\n",
      "\t511 of 33194 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t32683 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP6_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 914\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 914\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 35 (avg size of targets)\n",
      "\tBackground files for 35 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 99 sequences from chr1\n",
      "\tExtracting 78 sequences from chr2\n",
      "\tExtracting 60 sequences from chr3\n",
      "\tExtracting 38 sequences from chr4\n",
      "\tExtracting 47 sequences from chr5\n",
      "\tExtracting 44 sequences from chr6\n",
      "\tExtracting 44 sequences from chr7\n",
      "\tExtracting 29 sequences from chr8\n",
      "\tExtracting 37 sequences from chr9\n",
      "\tExtracting 33 sequences from chr10\n",
      "\tExtracting 58 sequences from chr11\n",
      "\tExtracting 45 sequences from chr12\n",
      "\tExtracting 12 sequences from chr13\n",
      "\tExtracting 31 sequences from chr14\n",
      "\tExtracting 36 sequences from chr15\n",
      "\tExtracting 40 sequences from chr16\n",
      "\tExtracting 50 sequences from chr17\n",
      "\tExtracting 19 sequences from chr18\n",
      "\tExtracting 35 sequences from chr19\n",
      "\tExtracting 26 sequences from chr20\n",
      "\tExtracting 10 sequences from chr21\n",
      "\tExtracting 14 sequences from chr22\n",
      "\tExtracting 23 sequences from chrX\n",
      "\tExtracting 6 sequences from chrY\n",
      "\t1 of 914 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t913 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP8_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 20495\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 20495\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 33 (avg size of targets)\n",
      "\tBackground files for 33 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 1923 sequences from chr1\n",
      "\tExtracting 1588 sequences from chr2\n",
      "\tExtracting 1209 sequences from chr3\n",
      "\tExtracting 900 sequences from chr4\n",
      "\tExtracting 998 sequences from chr5\n",
      "\tExtracting 974 sequences from chr6\n",
      "\tExtracting 1024 sequences from chr7\n",
      "\tExtracting 760 sequences from chr8\n",
      "\tExtracting 767 sequences from chr9\n",
      "\tExtracting 871 sequences from chr10\n",
      "\tExtracting 1089 sequences from chr11\n",
      "\tExtracting 1171 sequences from chr12\n",
      "\tExtracting 377 sequences from chr13\n",
      "\tExtracting 646 sequences from chr14\n",
      "\tExtracting 793 sequences from chr15\n",
      "\tExtracting 928 sequences from chr16\n",
      "\tExtracting 1114 sequences from chr17\n",
      "\tExtracting 351 sequences from chr18\n",
      "\tExtracting 1106 sequences from chr19\n",
      "\tExtracting 470 sequences from chr20\n",
      "\tExtracting 231 sequences from chr21\n",
      "\tExtracting 458 sequences from chr22\n",
      "\tExtracting 634 sequences from chrX\n",
      "\tExtracting 113 sequences from chrY\n",
      "\t277 of 20495 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t20218 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to200_gel_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 22370\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 22370\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 36 (avg size of targets)\n",
      "\tBackground files for 36 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2072 sequences from chr1\n",
      "\tExtracting 1682 sequences from chr2\n",
      "\tExtracting 1286 sequences from chr3\n",
      "\tExtracting 964 sequences from chr4\n",
      "\tExtracting 1085 sequences from chr5\n",
      "\tExtracting 1065 sequences from chr6\n",
      "\tExtracting 1157 sequences from chr7\n",
      "\tExtracting 866 sequences from chr8\n",
      "\tExtracting 862 sequences from chr9\n",
      "\tExtracting 929 sequences from chr10\n",
      "\tExtracting 1201 sequences from chr11\n",
      "\tExtracting 1277 sequences from chr12\n",
      "\tExtracting 418 sequences from chr13\n",
      "\tExtracting 736 sequences from chr14\n",
      "\tExtracting 860 sequences from chr15\n",
      "\tExtracting 941 sequences from chr16\n",
      "\tExtracting 1334 sequences from chr17\n",
      "\tExtracting 364 sequences from chr18\n",
      "\tExtracting 1189 sequences from chr19\n",
      "\tExtracting 542 sequences from chr20\n",
      "\tExtracting 231 sequences from chr21\n",
      "\tExtracting 505 sequences from chr22\n",
      "\tExtracting 708 sequences from chrX\n",
      "\tExtracting 96 sequences from chrY\n",
      "\t265 of 22370 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t22105 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to100_gel_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 34934\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 34934\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 38 (avg size of targets)\n",
      "\tBackground files for 38 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 3296 sequences from chr1\n",
      "\tExtracting 2644 sequences from chr2\n",
      "\tExtracting 2169 sequences from chr3\n",
      "\tExtracting 1459 sequences from chr4\n",
      "\tExtracting 1595 sequences from chr5\n",
      "\tExtracting 1718 sequences from chr6\n",
      "\tExtracting 1746 sequences from chr7\n",
      "\tExtracting 1372 sequences from chr8\n",
      "\tExtracting 1279 sequences from chr9\n",
      "\tExtracting 1469 sequences from chr10\n",
      "\tExtracting 1908 sequences from chr11\n",
      "\tExtracting 1981 sequences from chr12\n",
      "\tExtracting 621 sequences from chr13\n",
      "\tExtracting 1211 sequences from chr14\n",
      "\tExtracting 1349 sequences from chr15\n",
      "\tExtracting 1505 sequences from chr16\n",
      "\tExtracting 1948 sequences from chr17\n",
      "\tExtracting 631 sequences from chr18\n",
      "\tExtracting 1833 sequences from chr19\n",
      "\tExtracting 796 sequences from chr20\n",
      "\tExtracting 422 sequences from chr21\n",
      "\tExtracting 725 sequences from chr22\n",
      "\tExtracting 1060 sequences from chrX\n",
      "\tExtracting 196 sequences from chrY\n",
      "\tExtracting 1 sequences from chrM\n",
      "\t419 of 34934 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t34515 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP1_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 27038\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 27038\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 31 (avg size of targets)\n",
      "\tBackground files for 31 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2509 sequences from chr1\n",
      "\tExtracting 2092 sequences from chr2\n",
      "\tExtracting 1682 sequences from chr3\n",
      "\tExtracting 1130 sequences from chr4\n",
      "\tExtracting 1334 sequences from chr5\n",
      "\tExtracting 1348 sequences from chr6\n",
      "\tExtracting 1348 sequences from chr7\n",
      "\tExtracting 1070 sequences from chr8\n",
      "\tExtracting 1061 sequences from chr9\n",
      "\tExtracting 1066 sequences from chr10\n",
      "\tExtracting 1346 sequences from chr11\n",
      "\tExtracting 1496 sequences from chr12\n",
      "\tExtracting 545 sequences from chr13\n",
      "\tExtracting 871 sequences from chr14\n",
      "\tExtracting 1026 sequences from chr15\n",
      "\tExtracting 1162 sequences from chr16\n",
      "\tExtracting 1474 sequences from chr17\n",
      "\tExtracting 495 sequences from chr18\n",
      "\tExtracting 1382 sequences from chr19\n",
      "\tExtracting 626 sequences from chr20\n",
      "\tExtracting 312 sequences from chr21\n",
      "\tExtracting 590 sequences from chr22\n",
      "\tExtracting 920 sequences from chrX\n",
      "\tExtracting 153 sequences from chrY\n",
      "\t255 of 27038 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t26783 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to12_gel_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 33665\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 33665\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 41 (avg size of targets)\n",
      "\tBackground files for 41 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 3122 sequences from chr1\n",
      "\tExtracting 2572 sequences from chr2\n",
      "\tExtracting 2058 sequences from chr3\n",
      "\tExtracting 1458 sequences from chr4\n",
      "\tExtracting 1623 sequences from chr5\n",
      "\tExtracting 1575 sequences from chr6\n",
      "\tExtracting 1677 sequences from chr7\n",
      "\tExtracting 1291 sequences from chr8\n",
      "\tExtracting 1360 sequences from chr9\n",
      "\tExtracting 1445 sequences from chr10\n",
      "\tExtracting 1761 sequences from chr11\n",
      "\tExtracting 1864 sequences from chr12\n",
      "\tExtracting 696 sequences from chr13\n",
      "\tExtracting 1064 sequences from chr14\n",
      "\tExtracting 1322 sequences from chr15\n",
      "\tExtracting 1463 sequences from chr16\n",
      "\tExtracting 1807 sequences from chr17\n",
      "\tExtracting 603 sequences from chr18\n",
      "\tExtracting 1806 sequences from chr19\n",
      "\tExtracting 778 sequences from chr20\n",
      "\tExtracting 384 sequences from chr21\n",
      "\tExtracting 700 sequences from chr22\n",
      "\tExtracting 1060 sequences from chrX\n",
      "\tExtracting 175 sequences from chrY\n",
      "\tExtracting 1 sequences from chrM\n",
      "\t277 of 33665 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t33388 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to50_gel_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 28458\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 28458\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 41 (avg size of targets)\n",
      "\tBackground files for 41 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2640 sequences from chr1\n",
      "\tExtracting 2141 sequences from chr2\n",
      "\tExtracting 1711 sequences from chr3\n",
      "\tExtracting 1263 sequences from chr4\n",
      "\tExtracting 1350 sequences from chr5\n",
      "\tExtracting 1359 sequences from chr6\n",
      "\tExtracting 1473 sequences from chr7\n",
      "\tExtracting 1131 sequences from chr8\n",
      "\tExtracting 1096 sequences from chr9\n",
      "\tExtracting 1212 sequences from chr10\n",
      "\tExtracting 1437 sequences from chr11\n",
      "\tExtracting 1639 sequences from chr12\n",
      "\tExtracting 531 sequences from chr13\n",
      "\tExtracting 938 sequences from chr14\n",
      "\tExtracting 1107 sequences from chr15\n",
      "\tExtracting 1221 sequences from chr16\n",
      "\tExtracting 1606 sequences from chr17\n",
      "\tExtracting 515 sequences from chr18\n",
      "\tExtracting 1517 sequences from chr19\n",
      "\tExtracting 655 sequences from chr20\n",
      "\tExtracting 299 sequences from chr21\n",
      "\tExtracting 612 sequences from chr22\n",
      "\tExtracting 867 sequences from chrX\n",
      "\tExtracting 136 sequences from chrY\n",
      "\tExtracting 2 sequences from chrM\n",
      "\t270 of 28458 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t28188 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/EVN_041315_CLIP5_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 20718\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 20718\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 33 (avg size of targets)\n",
      "\tBackground files for 33 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 1855 sequences from chr1\n",
      "\tExtracting 1604 sequences from chr2\n",
      "\tExtracting 1266 sequences from chr3\n",
      "\tExtracting 992 sequences from chr4\n",
      "\tExtracting 1041 sequences from chr5\n",
      "\tExtracting 977 sequences from chr6\n",
      "\tExtracting 1106 sequences from chr7\n",
      "\tExtracting 807 sequences from chr8\n",
      "\tExtracting 836 sequences from chr9\n",
      "\tExtracting 931 sequences from chr10\n",
      "\tExtracting 1039 sequences from chr11\n",
      "\tExtracting 1122 sequences from chr12\n",
      "\tExtracting 430 sequences from chr13\n",
      "\tExtracting 650 sequences from chr14\n",
      "\tExtracting 815 sequences from chr15\n",
      "\tExtracting 879 sequences from chr16\n",
      "\tExtracting 1064 sequences from chr17\n",
      "\tExtracting 381 sequences from chr18\n",
      "\tExtracting 1018 sequences from chr19\n",
      "\tExtracting 506 sequences from chr20\n",
      "\tExtracting 198 sequences from chr21\n",
      "\tExtracting 420 sequences from chr22\n",
      "\tExtracting 647 sequences from chrX\n",
      "\tExtracting 133 sequences from chrY\n",
      "\tExtracting 1 sequences from chrM\n",
      "\t118 of 20718 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t20600 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = /projects/ps-yeolab2/encode/analysis/encode_v12/assigned/RBFOX2_293XT_1to25_gel_RBFOX2.merged.r2.peaks.bed.all.rand.0.BED\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = foo\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tOperating in RNA mode\n",
      "\tWill find motif(s) in /home/gpratt/projects/encode/analysis/encode_v8/homer/293XT_CLIP_RBFOX2_1120_RBFOX2.merged.r2.peaks.bed/all/homerResults/motif1.motif\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 27088\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 27088\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 38 (avg size of targets)\n",
      "\tBackground files for 38 bp fragments found.\n",
      "\n",
      "\tExtracting sequences from directory: /home/yeo-lab/software/homer/.//data/genomes/hg19//\n",
      "\tExtracting 2533 sequences from chr1\n",
      "\tExtracting 2136 sequences from chr2\n",
      "\tExtracting 1673 sequences from chr3\n",
      "\tExtracting 1119 sequences from chr4\n",
      "\tExtracting 1309 sequences from chr5\n",
      "\tExtracting 1240 sequences from chr6\n",
      "\tExtracting 1390 sequences from chr7\n",
      "\tExtracting 1040 sequences from chr8\n",
      "\tExtracting 1066 sequences from chr9\n",
      "\tExtracting 1101 sequences from chr10\n",
      "\tExtracting 1441 sequences from chr11\n",
      "\tExtracting 1436 sequences from chr12\n",
      "\tExtracting 527 sequences from chr13\n",
      "\tExtracting 878 sequences from chr14\n",
      "\tExtracting 1035 sequences from chr15\n",
      "\tExtracting 1136 sequences from chr16\n",
      "\tExtracting 1494 sequences from chr17\n",
      "\tExtracting 465 sequences from chr18\n",
      "\tExtracting 1472 sequences from chr19\n",
      "\tExtracting 656 sequences from chr20\n",
      "\tExtracting 308 sequences from chr21\n",
      "\tExtracting 592 sequences from chr22\n",
      "\tExtracting 857 sequences from chrX\n",
      "\tExtracting 184 sequences from chrY\n",
      "\t239 of 27088 removed because they were shorter than 5 bp\n",
      "\n",
      "\n",
      "\tReading input files...\n",
      "\t26849 total sequences read\n",
      "\t1 motifs loaded\n",
      "\tFinding instances of 1 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tCleaning up tmp files...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "results = {}\n",
    "for peak in peaks_to_annotate:\n",
    "    annotated_peak = findMotifsGenome(peaks_to_annotate[peak], rbfox2_motif)\n",
    "    annotated_peak['bound_in_same_strand'] = annotated_peak.apply(bound_in_same_strand, axis=1)\n",
    "    results[peak] = {\"peaks_with_motif\": len(set(annotated_peak.index)), \"num_peaks\": len(pybedtools.BedTool(peaks_to_annotate[peak]))}\n",
    "\n",
    "results_rand_1 = {}\n",
    "for peak in peaks_to_rand_1_annotate:\n",
    "    annotated_peak = findMotifsGenome(peaks_to_rand_1_annotate[peak], rbfox2_motif)\n",
    "    annotated_peak['bound_in_same_strand'] = annotated_peak.apply(bound_in_same_strand, axis=1)\n",
    "    results_rand_1[peak] = {\"peaks_with_motif\": len(set(annotated_peak.index)), \"num_peaks\": len(pybedtools.BedTool(peaks_to_annotate[peak]))}\n",
    "    \n",
    "results_rand_2 = {}\n",
    "for peak in peaks_to_rand_2_annotate:\n",
    "    annotated_peak = findMotifsGenome(peaks_to_rand_2_annotate[peak], rbfox2_motif)\n",
    "    annotated_peak['bound_in_same_strand'] = annotated_peak.apply(bound_in_same_strand, axis=1)\n",
    "    results_rand_2[peak] = {\"peaks_with_motif\": len(set(annotated_peak.index)), \"num_peaks\": len(pybedtools.BedTool(peaks_to_annotate[peak]))}\n",
    "    \n",
    "results_rand_3 = {}\n",
    "for peak in peaks_to_rand_3_annotate:\n",
    "    annotated_peak = findMotifsGenome(peaks_to_rand_3_annotate[peak], rbfox2_motif)\n",
    "    annotated_peak['bound_in_same_strand'] = annotated_peak.apply(bound_in_same_strand, axis=1)\n",
    "    results_rand_3[peak] = {\"peaks_with_motif\": len(set(annotated_peak.index)), \"num_peaks\": len(pybedtools.BedTool(peaks_to_annotate[peak]))}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "foo = pybedtools.BedTool(peaks_to_annotate[peak]).filter(lambda interval: interval.name in set(annotated_peak.index) - found).saveas()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Offset                                                                 51\n",
       "Sequence                                                           TGCATG\n",
       "Motif Name              1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...\n",
       "Strand                                                                  +\n",
       "MotifScore                                                        8.24193\n",
       "bound_in_same_strand                                                False\n",
       "Name: ENSG00000078900.10_1_42_262, dtype: object"
      ]
     },
     "execution_count": 173,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_peak.ix['ENSG00000078900.10_1_42_262']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Offset</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Motif Name</th>\n",
       "      <th>Strand</th>\n",
       "      <th>MotifScore</th>\n",
       "      <th>bound_in_same_strand</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PositionID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000105983.14_4_9_26031</th>\n",
       "      <td>2</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000159720.7_1_9_10474</th>\n",
       "      <td>20</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000031823.10_5_29_14272</th>\n",
       "      <td>9</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000092439.9_0_9_8522</th>\n",
       "      <td>0</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000135476.7_11_8_6130</th>\n",
       "      <td>11</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000185483.7_1_5_1372</th>\n",
       "      <td>1</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000198952.6_4_13_2140</th>\n",
       "      <td>8</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000163931.11_5_15_21041</th>\n",
       "      <td>9</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000009954.6_2_7_25260</th>\n",
       "      <td>9</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000181220.11_5_5_25863</th>\n",
       "      <td>18</td>\n",
       "      <td>CGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>5.438040</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000173818.12_5_18_13188</th>\n",
       "      <td>20</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000095787.16_7_5_3215</th>\n",
       "      <td>9</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000250337.1_0_4_22528</th>\n",
       "      <td>1</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000116604.12_0_15_2153</th>\n",
       "      <td>1</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000170776.15_2_3_8942</th>\n",
       "      <td>1</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000175115.7_2_11_4996</th>\n",
       "      <td>1</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000101190.8_1_7_18441</th>\n",
       "      <td>7</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000139131.8_0_13_5922</th>\n",
       "      <td>34</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000178234.8_1_6_25974</th>\n",
       "      <td>0</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000100764.9_0_3_8021</th>\n",
       "      <td>8</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000126106.9_1_4_1184</th>\n",
       "      <td>1</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000111358.9_7_35_6953</th>\n",
       "      <td>5</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000100852.8_2_9_7689</th>\n",
       "      <td>15</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000173757.4_1_4_12204</th>\n",
       "      <td>10</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000163867.11_1_7_939</th>\n",
       "      <td>27</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000134769.17_4_5_13613</th>\n",
       "      <td>0</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000136699.15_11_161_17020,ENSG00000136699.15_12_158_17021,ENSG00000136699.15_13_81_17022</th>\n",
       "      <td>73</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000170085.13_2_13_23199</th>\n",
       "      <td>20</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000149548.10_2_21_5549</th>\n",
       "      <td>5</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000151576.6_0_6_21219</th>\n",
       "      <td>1</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000139629.11_0_3_6093</th>\n",
       "      <td>2</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000204681.6_0_10_23721</th>\n",
       "      <td>0</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000204681.6_0_10_23721</th>\n",
       "      <td>36</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000164051.9_1_8_20653</th>\n",
       "      <td>30</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000011405.9_1_4_4273</th>\n",
       "      <td>17</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000160439.10_2_4_15951</th>\n",
       "      <td>29</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000160439.10_2_4_15951</th>\n",
       "      <td>33</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000108264.12_1_4_12015</th>\n",
       "      <td>1</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000225032.1_0_5_27493</th>\n",
       "      <td>9</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000060339.9_2_6_3325</th>\n",
       "      <td>0</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000106603.13_7_18_25057</th>\n",
       "      <td>4</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000103365.11_3_29_9942</th>\n",
       "      <td>3</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000076555.11_3_6_6574</th>\n",
       "      <td>15</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000007168.7_0_7_11161</th>\n",
       "      <td>38</td>\n",
       "      <td>CGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>5.438040</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000082898.12_3_13_16511</th>\n",
       "      <td>6</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000173473.6_3_12_20611</th>\n",
       "      <td>3</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000119514.5_0_8_27309</th>\n",
       "      <td>16</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000119514.5_0_8_27309</th>\n",
       "      <td>30</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000165626.12_9_6_3154</th>\n",
       "      <td>18</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000082898.12_1_13_16509</th>\n",
       "      <td>13</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000114388.8_3_43_20856</th>\n",
       "      <td>15</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000154380.11_4_48_2657</th>\n",
       "      <td>3</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000044459.10_1_6_26925</th>\n",
       "      <td>20</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000165322.13_0_3_3223</th>\n",
       "      <td>2</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000183723.8_4_4_10406</th>\n",
       "      <td>1</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000075643.5_2_9_13643</th>\n",
       "      <td>8</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000010017.9_1_4_23623</th>\n",
       "      <td>14</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000123739.6_0_4_22244</th>\n",
       "      <td>1</td>\n",
       "      <td>CGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>5.438040</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000158467.12_0_8_25683</th>\n",
       "      <td>5</td>\n",
       "      <td>TGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>8.241931</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000149541.5_3_38_4630</th>\n",
       "      <td>0</td>\n",
       "      <td>CGCATG</td>\n",
       "      <td>1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...</td>\n",
       "      <td>+</td>\n",
       "      <td>5.438040</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>357 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Offset Sequence  \\\n",
       "PositionID                                                            \n",
       "ENSG00000105983.14_4_9_26031                             2   TGCATG   \n",
       "ENSG00000159720.7_1_9_10474                             20   TGCATG   \n",
       "ENSG00000031823.10_5_29_14272                            9   TGCATG   \n",
       "ENSG00000092439.9_0_9_8522                               0   TGCATG   \n",
       "ENSG00000135476.7_11_8_6130                             11   TGCATG   \n",
       "ENSG00000185483.7_1_5_1372                               1   TGCATG   \n",
       "ENSG00000198952.6_4_13_2140                              8   TGCATG   \n",
       "ENSG00000163931.11_5_15_21041                            9   TGCATG   \n",
       "ENSG00000009954.6_2_7_25260                              9   TGCATG   \n",
       "ENSG00000181220.11_5_5_25863                            18   CGCATG   \n",
       "ENSG00000173818.12_5_18_13188                           20   TGCATG   \n",
       "ENSG00000095787.16_7_5_3215                              9   TGCATG   \n",
       "ENSG00000250337.1_0_4_22528                              1   TGCATG   \n",
       "ENSG00000116604.12_0_15_2153                             1   TGCATG   \n",
       "ENSG00000170776.15_2_3_8942                              1   TGCATG   \n",
       "ENSG00000175115.7_2_11_4996                              1   TGCATG   \n",
       "ENSG00000101190.8_1_7_18441                              7   TGCATG   \n",
       "ENSG00000139131.8_0_13_5922                             34   TGCATG   \n",
       "ENSG00000178234.8_1_6_25974                              0   TGCATG   \n",
       "ENSG00000100764.9_0_3_8021                               8   TGCATG   \n",
       "ENSG00000126106.9_1_4_1184                               1   TGCATG   \n",
       "ENSG00000111358.9_7_35_6953                              5   TGCATG   \n",
       "ENSG00000100852.8_2_9_7689                              15   TGCATG   \n",
       "ENSG00000173757.4_1_4_12204                             10   TGCATG   \n",
       "ENSG00000163867.11_1_7_939                              27   TGCATG   \n",
       "ENSG00000134769.17_4_5_13613                             0   TGCATG   \n",
       "ENSG00000136699.15_11_161_17020,ENSG00000136699...      73   TGCATG   \n",
       "ENSG00000170085.13_2_13_23199                           20   TGCATG   \n",
       "ENSG00000149548.10_2_21_5549                             5   TGCATG   \n",
       "ENSG00000151576.6_0_6_21219                              1   TGCATG   \n",
       "...                                                    ...      ...   \n",
       "ENSG00000139629.11_0_3_6093                              2   TGCATG   \n",
       "ENSG00000204681.6_0_10_23721                             0   TGCATG   \n",
       "ENSG00000204681.6_0_10_23721                            36   TGCATG   \n",
       "ENSG00000164051.9_1_8_20653                             30   TGCATG   \n",
       "ENSG00000011405.9_1_4_4273                              17   TGCATG   \n",
       "ENSG00000160439.10_2_4_15951                            29   TGCATG   \n",
       "ENSG00000160439.10_2_4_15951                            33   TGCATG   \n",
       "ENSG00000108264.12_1_4_12015                             1   TGCATG   \n",
       "ENSG00000225032.1_0_5_27493                              9   TGCATG   \n",
       "ENSG00000060339.9_2_6_3325                               0   TGCATG   \n",
       "ENSG00000106603.13_7_18_25057                            4   TGCATG   \n",
       "ENSG00000103365.11_3_29_9942                             3   TGCATG   \n",
       "ENSG00000076555.11_3_6_6574                             15   TGCATG   \n",
       "ENSG00000007168.7_0_7_11161                             38   CGCATG   \n",
       "ENSG00000082898.12_3_13_16511                            6   TGCATG   \n",
       "ENSG00000173473.6_3_12_20611                             3   TGCATG   \n",
       "ENSG00000119514.5_0_8_27309                             16   TGCATG   \n",
       "ENSG00000119514.5_0_8_27309                             30   TGCATG   \n",
       "ENSG00000165626.12_9_6_3154                             18   TGCATG   \n",
       "ENSG00000082898.12_1_13_16509                           13   TGCATG   \n",
       "ENSG00000114388.8_3_43_20856                            15   TGCATG   \n",
       "ENSG00000154380.11_4_48_2657                             3   TGCATG   \n",
       "ENSG00000044459.10_1_6_26925                            20   TGCATG   \n",
       "ENSG00000165322.13_0_3_3223                              2   TGCATG   \n",
       "ENSG00000183723.8_4_4_10406                              1   TGCATG   \n",
       "ENSG00000075643.5_2_9_13643                              8   TGCATG   \n",
       "ENSG00000010017.9_1_4_23623                             14   TGCATG   \n",
       "ENSG00000123739.6_0_4_22244                              1   CGCATG   \n",
       "ENSG00000158467.12_0_8_25683                             5   TGCATG   \n",
       "ENSG00000149541.5_3_38_4630                              0   CGCATG   \n",
       "\n",
       "                                                                                           Motif Name  \\\n",
       "PositionID                                                                                              \n",
       "ENSG00000105983.14_4_9_26031                        1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000159720.7_1_9_10474                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000031823.10_5_29_14272                       1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000092439.9_0_9_8522                          1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000135476.7_11_8_6130                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000185483.7_1_5_1372                          1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000198952.6_4_13_2140                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000163931.11_5_15_21041                       1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000009954.6_2_7_25260                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000181220.11_5_5_25863                        1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000173818.12_5_18_13188                       1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000095787.16_7_5_3215                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000250337.1_0_4_22528                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000116604.12_0_15_2153                        1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000170776.15_2_3_8942                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000175115.7_2_11_4996                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000101190.8_1_7_18441                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000139131.8_0_13_5922                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000178234.8_1_6_25974                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000100764.9_0_3_8021                          1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000126106.9_1_4_1184                          1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000111358.9_7_35_6953                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000100852.8_2_9_7689                          1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000173757.4_1_4_12204                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000163867.11_1_7_939                          1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000134769.17_4_5_13613                        1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000136699.15_11_161_17020,ENSG00000136699...  1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000170085.13_2_13_23199                       1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000149548.10_2_21_5549                        1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000151576.6_0_6_21219                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "...                                                                                               ...   \n",
       "ENSG00000139629.11_0_3_6093                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000204681.6_0_10_23721                        1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000204681.6_0_10_23721                        1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000164051.9_1_8_20653                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000011405.9_1_4_4273                          1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000160439.10_2_4_15951                        1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000160439.10_2_4_15951                        1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000108264.12_1_4_12015                        1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000225032.1_0_5_27493                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000060339.9_2_6_3325                          1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000106603.13_7_18_25057                       1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000103365.11_3_29_9942                        1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000076555.11_3_6_6574                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000007168.7_0_7_11161                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000082898.12_3_13_16511                       1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000173473.6_3_12_20611                        1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000119514.5_0_8_27309                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000119514.5_0_8_27309                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000165626.12_9_6_3154                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000082898.12_1_13_16509                       1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000114388.8_3_43_20856                        1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000154380.11_4_48_2657                        1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000044459.10_1_6_26925                        1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000165322.13_0_3_3223                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000183723.8_4_4_10406                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000075643.5_2_9_13643                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000010017.9_1_4_23623                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000123739.6_0_4_22244                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000158467.12_0_8_25683                        1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "ENSG00000149541.5_3_38_4630                         1-TGCATG,BestGuess:A2BP1_ENSG00000078328_M298_...   \n",
       "\n",
       "                                                   Strand  MotifScore  \\\n",
       "PositionID                                                              \n",
       "ENSG00000105983.14_4_9_26031                            +    8.241931   \n",
       "ENSG00000159720.7_1_9_10474                             +    8.241931   \n",
       "ENSG00000031823.10_5_29_14272                           +    8.241931   \n",
       "ENSG00000092439.9_0_9_8522                              +    8.241931   \n",
       "ENSG00000135476.7_11_8_6130                             +    8.241931   \n",
       "ENSG00000185483.7_1_5_1372                              +    8.241931   \n",
       "ENSG00000198952.6_4_13_2140                             +    8.241931   \n",
       "ENSG00000163931.11_5_15_21041                           +    8.241931   \n",
       "ENSG00000009954.6_2_7_25260                             +    8.241931   \n",
       "ENSG00000181220.11_5_5_25863                            +    5.438040   \n",
       "ENSG00000173818.12_5_18_13188                           +    8.241931   \n",
       "ENSG00000095787.16_7_5_3215                             +    8.241931   \n",
       "ENSG00000250337.1_0_4_22528                             +    8.241931   \n",
       "ENSG00000116604.12_0_15_2153                            +    8.241931   \n",
       "ENSG00000170776.15_2_3_8942                             +    8.241931   \n",
       "ENSG00000175115.7_2_11_4996                             +    8.241931   \n",
       "ENSG00000101190.8_1_7_18441                             +    8.241931   \n",
       "ENSG00000139131.8_0_13_5922                             +    8.241931   \n",
       "ENSG00000178234.8_1_6_25974                             +    8.241931   \n",
       "ENSG00000100764.9_0_3_8021                              +    8.241931   \n",
       "ENSG00000126106.9_1_4_1184                              +    8.241931   \n",
       "ENSG00000111358.9_7_35_6953                             +    8.241931   \n",
       "ENSG00000100852.8_2_9_7689                              +    8.241931   \n",
       "ENSG00000173757.4_1_4_12204                             +    8.241931   \n",
       "ENSG00000163867.11_1_7_939                              +    8.241931   \n",
       "ENSG00000134769.17_4_5_13613                            +    8.241931   \n",
       "ENSG00000136699.15_11_161_17020,ENSG00000136699...      +    8.241931   \n",
       "ENSG00000170085.13_2_13_23199                           +    8.241931   \n",
       "ENSG00000149548.10_2_21_5549                            +    8.241931   \n",
       "ENSG00000151576.6_0_6_21219                             +    8.241931   \n",
       "...                                                   ...         ...   \n",
       "ENSG00000139629.11_0_3_6093                             +    8.241931   \n",
       "ENSG00000204681.6_0_10_23721                            +    8.241931   \n",
       "ENSG00000204681.6_0_10_23721                            +    8.241931   \n",
       "ENSG00000164051.9_1_8_20653                             +    8.241931   \n",
       "ENSG00000011405.9_1_4_4273                              +    8.241931   \n",
       "ENSG00000160439.10_2_4_15951                            +    8.241931   \n",
       "ENSG00000160439.10_2_4_15951                            +    8.241931   \n",
       "ENSG00000108264.12_1_4_12015                            +    8.241931   \n",
       "ENSG00000225032.1_0_5_27493                             +    8.241931   \n",
       "ENSG00000060339.9_2_6_3325                              +    8.241931   \n",
       "ENSG00000106603.13_7_18_25057                           +    8.241931   \n",
       "ENSG00000103365.11_3_29_9942                            +    8.241931   \n",
       "ENSG00000076555.11_3_6_6574                             +    8.241931   \n",
       "ENSG00000007168.7_0_7_11161                             +    5.438040   \n",
       "ENSG00000082898.12_3_13_16511                           +    8.241931   \n",
       "ENSG00000173473.6_3_12_20611                            +    8.241931   \n",
       "ENSG00000119514.5_0_8_27309                             +    8.241931   \n",
       "ENSG00000119514.5_0_8_27309                             +    8.241931   \n",
       "ENSG00000165626.12_9_6_3154                             +    8.241931   \n",
       "ENSG00000082898.12_1_13_16509                           +    8.241931   \n",
       "ENSG00000114388.8_3_43_20856                            +    8.241931   \n",
       "ENSG00000154380.11_4_48_2657                            +    8.241931   \n",
       "ENSG00000044459.10_1_6_26925                            +    8.241931   \n",
       "ENSG00000165322.13_0_3_3223                             +    8.241931   \n",
       "ENSG00000183723.8_4_4_10406                             +    8.241931   \n",
       "ENSG00000075643.5_2_9_13643                             +    8.241931   \n",
       "ENSG00000010017.9_1_4_23623                             +    8.241931   \n",
       "ENSG00000123739.6_0_4_22244                             +    5.438040   \n",
       "ENSG00000158467.12_0_8_25683                            +    8.241931   \n",
       "ENSG00000149541.5_3_38_4630                             +    5.438040   \n",
       "\n",
       "                                                   bound_in_same_strand  \n",
       "PositionID                                                               \n",
       "ENSG00000105983.14_4_9_26031                                      False  \n",
       "ENSG00000159720.7_1_9_10474                                       False  \n",
       "ENSG00000031823.10_5_29_14272                                     False  \n",
       "ENSG00000092439.9_0_9_8522                                        False  \n",
       "ENSG00000135476.7_11_8_6130                                       False  \n",
       "ENSG00000185483.7_1_5_1372                                        False  \n",
       "ENSG00000198952.6_4_13_2140                                       False  \n",
       "ENSG00000163931.11_5_15_21041                                     False  \n",
       "ENSG00000009954.6_2_7_25260                                       False  \n",
       "ENSG00000181220.11_5_5_25863                                      False  \n",
       "ENSG00000173818.12_5_18_13188                                     False  \n",
       "ENSG00000095787.16_7_5_3215                                       False  \n",
       "ENSG00000250337.1_0_4_22528                                       False  \n",
       "ENSG00000116604.12_0_15_2153                                      False  \n",
       "ENSG00000170776.15_2_3_8942                                       False  \n",
       "ENSG00000175115.7_2_11_4996                                       False  \n",
       "ENSG00000101190.8_1_7_18441                                       False  \n",
       "ENSG00000139131.8_0_13_5922                                       False  \n",
       "ENSG00000178234.8_1_6_25974                                       False  \n",
       "ENSG00000100764.9_0_3_8021                                        False  \n",
       "ENSG00000126106.9_1_4_1184                                        False  \n",
       "ENSG00000111358.9_7_35_6953                                       False  \n",
       "ENSG00000100852.8_2_9_7689                                        False  \n",
       "ENSG00000173757.4_1_4_12204                                       False  \n",
       "ENSG00000163867.11_1_7_939                                        False  \n",
       "ENSG00000134769.17_4_5_13613                                      False  \n",
       "ENSG00000136699.15_11_161_17020,ENSG00000136699...                False  \n",
       "ENSG00000170085.13_2_13_23199                                     False  \n",
       "ENSG00000149548.10_2_21_5549                                      False  \n",
       "ENSG00000151576.6_0_6_21219                                       False  \n",
       "...                                                                 ...  \n",
       "ENSG00000139629.11_0_3_6093                                       False  \n",
       "ENSG00000204681.6_0_10_23721                                      False  \n",
       "ENSG00000204681.6_0_10_23721                                      False  \n",
       "ENSG00000164051.9_1_8_20653                                       False  \n",
       "ENSG00000011405.9_1_4_4273                                        False  \n",
       "ENSG00000160439.10_2_4_15951                                      False  \n",
       "ENSG00000160439.10_2_4_15951                                      False  \n",
       "ENSG00000108264.12_1_4_12015                                      False  \n",
       "ENSG00000225032.1_0_5_27493                                       False  \n",
       "ENSG00000060339.9_2_6_3325                                        False  \n",
       "ENSG00000106603.13_7_18_25057                                     False  \n",
       "ENSG00000103365.11_3_29_9942                                      False  \n",
       "ENSG00000076555.11_3_6_6574                                       False  \n",
       "ENSG00000007168.7_0_7_11161                                       False  \n",
       "ENSG00000082898.12_3_13_16511                                     False  \n",
       "ENSG00000173473.6_3_12_20611                                      False  \n",
       "ENSG00000119514.5_0_8_27309                                       False  \n",
       "ENSG00000119514.5_0_8_27309                                       False  \n",
       "ENSG00000165626.12_9_6_3154                                       False  \n",
       "ENSG00000082898.12_1_13_16509                                     False  \n",
       "ENSG00000114388.8_3_43_20856                                      False  \n",
       "ENSG00000154380.11_4_48_2657                                      False  \n",
       "ENSG00000044459.10_1_6_26925                                      False  \n",
       "ENSG00000165322.13_0_3_3223                                       False  \n",
       "ENSG00000183723.8_4_4_10406                                       False  \n",
       "ENSG00000075643.5_2_9_13643                                       False  \n",
       "ENSG00000010017.9_1_4_23623                                       False  \n",
       "ENSG00000123739.6_0_4_22244                                       False  \n",
       "ENSG00000158467.12_0_8_25683                                      False  \n",
       "ENSG00000149541.5_3_38_4630                                       False  \n",
       "\n",
       "[357 rows x 6 columns]"
      ]
     },
     "execution_count": 164,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_peak.ix[set(annotated_peak.index) - found]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "foo = pybedtools.BedTool(peaks_to_annotate[peak]).filter(lambda interval: not interval.name in set(annotated_peak_v2.index)).saveas()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "fasta_sequence = pybedtools.BedTool(peaks_to_annotate[peak]).sequence(fi=\"/projects/ps-yeolab/genomes/hg19/chromosomes/all.fa\", name=True, fo=\"foo.fasta\", s=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "found_motif = 0\n",
    "found = set([])\n",
    "for record in SeqIO.parse(\"foo.fasta\", format=\"fasta\"):\n",
    "    if \"CGCATG\".lower() in record.seq.lower() or \"TGCATG\".lower() in record.seq.lower():\n",
    "        found_motif += 1\n",
    "        found.add(record.name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "real = pd.DataFrame(results).T\n",
    "\n",
    "random = pd.concat([pd.DataFrame(results_rand_1).T, \n",
    "           pd.DataFrame(results_rand_2).T, \n",
    "           pd.DataFrame(results_rand_3).T,])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "real.to_csv(\"/home/gpratt/Dropbox/encode_integration/for_eric/real_counts_v12.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "random.groupby(level=0).sum().to_csv(\"/home/gpratt/Dropbox/encode_integration/for_eric/random_counts_v12.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
